Archaic humans - SNPs found in NH and DH
================

Set the environment and creating metadata file.

``` r
library(tidyverse)
```

    ## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
    ## ✔ dplyr     1.1.0     ✔ readr     2.1.4
    ## ✔ forcats   1.0.0     ✔ stringr   1.5.0
    ## ✔ ggplot2   3.4.1     ✔ tibble    3.2.0
    ## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
    ## ✔ purrr     1.0.1     
    ## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()
    ## ℹ Use the ]8;;http://conflicted.r-lib.org/conflicted package]8;; to force all conflicts to become errors

``` r
library(umap)

HGDP <- read_delim("/Volumes/Temp1/rpianezza/0.old/summary-HGDP/HGDP_cutoff_classified.tsv") %>% select(ID, sex, country, pop) %>% distinct()
```

    ## Rows: 1394352 Columns: 12
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (9): ID, pop, sex, country, type, familyname, batch, superfamily, shared...
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
archaic <- tibble(
  ID = c("Vindija", "Altai", "Denisovan"),
  pop = c("Vindija", "Altai", "Denisova"),
  country = c("Neandertal", "Neandertal", "Denisovan"),
  sex = c("female", "female", "female")
)

HGDP_arch <- bind_rows(HGDP, archaic)

coordinates <- read_tsv("/Users/rpianezza/TE/summary-HGDP/HGDP_populationcoordinates.txt", col_names = c("pop", "region", "latitude", "longitude")) %>% select(pop, latitude, longitude)
```

    ## Rows: 54 Columns: 4
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (2): pop, region
    ## dbl (2): latitude, longitude
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
afr <- HGDP %>% filter(country=="Africa", sex=="female") %>% select(ID)
#write_tsv(afr, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/ID-Afr-females")

fem <- HGDP %>% filter(sex=="female") %>% select(ID)
#write_tsv(fem, "/Volumes/Temp1/rpianezza/PCA-SNPs-all-analysis/matrixes/females/females.IDs")

mal <- HGDP %>% filter(sex=="male") %>% select(ID)
#write_tsv(mal, "/Volumes/Temp1/rpianezza/PCA-SNPs-all-analysis/matrixes/females/males.IDs")
```

## SCGs

Reading the matrixes for the archaic variants in SCGs in the HGDP. 1
file for each archaic genome.

``` r
scg_vindija_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/vindija1k")
```

    ## Rows: 1000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
scg_altai_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/altai1k")
```

    ## Rows: 1000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
scg_denisova_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/denisova1k")
```

    ## Rows: 1000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

Read the files containing info about the variants called in the archaic
genome. Is important the “total_diff” column, which is an indicator of
the distance between the allele frequencies in the Africans and in the
archaic genome.

Is comprised between 0 (same allele frequencies for that position) and 2
(complete non-overlapping allele frequencies for that position).

``` r
scg_diff_vindija <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/vindija1k_afr-arch") %>% select(familyname, position, total_diff)
```

    ## Rows: 1000 Columns: 15
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (14): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, A_diff, T_diff, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
scg_diff_altai <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/altai1k_afr-arch") %>% select(familyname, position, total_diff)
```

    ## Rows: 1000 Columns: 15
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (14): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, A_diff, T_diff, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
scg_diff_denisova <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/denisova1k_afr-arch") %>% select(familyname, position, total_diff)
```

    ## Rows: 1000 Columns: 15
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (14): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, A_diff, T_diff, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

Merge the two files read above for each of the three genome and select
only the variance with a **total_diff** \> a certain threshold. In this
case, I use 1.75. This means that, for single copy genes, both the
alleles must be different between archaic and africans to keep the
variant in this code. Ideally, this will result in a total_diff = 2, but
due to sequencing error and aDNA bias I use a lower threshold (1.97).

``` r
scg_vindija <- inner_join(scg_vindija_all, scg_diff_vindija, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
scg_altai <- inner_join(scg_altai_all, scg_diff_altai, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
scg_denisova <- inner_join(scg_denisova_all, scg_diff_denisova, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
```

With the selected variants, I then look for potentially introgressed
variants in all the HGDP samples: the variants which are very different
between archaic and africans (total_diff very high) but that in a genome
are not that different. I use a threshold of **total_diff** of 1.25 to
select those variant which are supposed to be introgressed in one copy
in a genome (ideally 1).

``` r
intro_vindija <- scg_vindija %>% select(-familyname, -position, -total_diff) %>% summarise_all(~sum(. <1.25, na.rm=TRUE)) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1") %>% as_tibble() %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP, by="ID")
intro_altai <- scg_altai %>% select(-familyname, -position, -total_diff) %>% summarise_all(~sum(. <1.25, na.rm=TRUE)) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1") %>% as_tibble() %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP, by="ID")
(intro_denisova <- scg_denisova %>% select(-familyname, -position, -total_diff) %>% summarise_all(~sum(. <1.25, na.rm=TRUE)) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1") %>% as_tibble() %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP, by="ID"))
```

    ## # A tibble: 828 × 5
    ##    ID        introgressed sex   country            pop   
    ##    <chr>            <int> <chr> <chr>              <chr> 
    ##  1 HGDP00001            6 male  Central_South_Asia Brahui
    ##  2 HGDP00003           21 male  Central_South_Asia Brahui
    ##  3 HGDP00005           25 male  Central_South_Asia Brahui
    ##  4 HGDP00007           30 male  Central_South_Asia Brahui
    ##  5 HGDP00009           19 male  Central_South_Asia Brahui
    ##  6 HGDP00011           23 male  Central_South_Asia Brahui
    ##  7 HGDP00013            2 male  Central_South_Asia Brahui
    ##  8 HGDP00015           25 male  Central_South_Asia Brahui
    ##  9 HGDP00017           25 male  Central_South_Asia Brahui
    ## 10 HGDP00021           25 male  Central_South_Asia Brahui
    ## # … with 818 more rows

From the info above, I produce these plots averaging the number of
introgressed variants for each population.

``` r
(scg_v_plot <- intro_vindija %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(introgressed), sd_intro = sd(introgressed)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro), 
                width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Introgressed variants from Vindija Neandertal found in SCGs")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/scg-plot-pop-1.png)<!-- -->

``` r
(scg_a_plot <- intro_altai %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(introgressed), sd_intro = sd(introgressed)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro), 
                width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Introgressed variants from Altai Neandertal found in SCGs")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/scg-plot-pop-2.png)<!-- -->

``` r
(scg_d_plot <- intro_denisova %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(introgressed), sd_intro = sd(introgressed)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro), 
                width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Introgressed variants from Denisovans found in SCGs")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/scg-plot-pop-3.png)<!-- -->

``` r
plot_map <- function(intro, coord, title){
  
full <- inner_join(intro, coord, by="pop") %>% group_by(pop, longitude, latitude) %>% dplyr::summarise(introgressed = mean(introgressed), count = n())
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(data = full, aes(x=longitude, y=latitude, color = introgressed, size = count)) + geom_errorbar() + scale_colour_gradient(low = "green", high = "red") + theme(legend.position="top") + theme(plot.title = element_text(hjust = 0.5)) + ggtitle(title)
}


plot_map(intro_vindija, coordinates, "Variants introgressed from Vindija NH found in SCGs")
```

    ## `summarise()` has grouped output by 'pop', 'longitude'. You can override using
    ## the `.groups` argument.

    ## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
    ## ℹ Please use `linewidth` instead.

    ## Warning in geom_map(data = world_map, map = world_map, aes(long, lat, map_id =
    ## region), : Ignoring unknown aesthetics: x and y

![](02.archaic-SNPs_files/figure-gfm/scg-map-1.png)<!-- -->

``` r
plot_map(intro_altai, coordinates, "Variants introgressed from Altai NH found in SCGs")
```

    ## `summarise()` has grouped output by 'pop', 'longitude'. You can override using
    ## the `.groups` argument.

    ## Warning in geom_map(data = world_map, map = world_map, aes(long, lat, map_id =
    ## region), : Ignoring unknown aesthetics: x and y

![](02.archaic-SNPs_files/figure-gfm/scg-map-2.png)<!-- -->

``` r
plot_map(intro_denisova, coordinates, "Variants introgressed from Denisovans found in SCGs")
```

    ## `summarise()` has grouped output by 'pop', 'longitude'. You can override using
    ## the `.groups` argument.

    ## Warning in geom_map(data = world_map, map = world_map, aes(long, lat, map_id =
    ## region), : Ignoring unknown aesthetics: x and y

![](02.archaic-SNPs_files/figure-gfm/scg-map-3.png)<!-- -->

Here I do the same for each region (“country”).

``` r
(scg_v_plot_cou <- intro_vindija %>% group_by(country) %>% dplyr::summarise(mean_intro = mean(introgressed)) %>% ggplot(aes(x = reorder(country, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Introgressed variants from Vindija Neandertal found in SCGs")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

![](02.archaic-SNPs_files/figure-gfm/scg-plot-country-1.png)<!-- -->

``` r
(scg_a_plot_cou <- intro_altai %>% group_by(country) %>% dplyr::summarise(mean_intro = mean(introgressed)) %>% ggplot(aes(x = reorder(country, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Introgressed variants from Altai Neandertal found in SCGs")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

![](02.archaic-SNPs_files/figure-gfm/scg-plot-country-2.png)<!-- -->

``` r
(scg_d_plot_cou <- intro_denisova %>% group_by(country) %>% dplyr::summarise(mean_intro = mean(introgressed)) %>% ggplot(aes(x = reorder(country, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Introgressed variants from Denisovans found in SCGs")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

![](02.archaic-SNPs_files/figure-gfm/scg-plot-country-3.png)<!-- -->

# TEs (variants selected using only African females)

``` r
HGDP_fem <- filter(HGDP, sex=="female") %>% select(ID, country, pop)

HGDP_pcr_free_samples <- read_tsv("/Volumes/Temp1/rpianezza/investigation/HGDP-no-PCR/HGDP-only-pcr-free-samples.tsv", col_names = "ID")
```

    ## Rows: 676 Columns: 1
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (1): ID
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
HGDP_pcr_free <- HGDP_fem %>% filter(ID %in% HGDP_pcr_free_samples$ID)

a_HGDP <- read_tsv("/Volumes/Temp1/rpianezza/PCA-copynumber-all-analysis/a_HGDP.tsv")
```

    ## Rows: 828 Columns: 2
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (1): ID
    ## dbl (1): a
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
HGDP_nobiased_samples <- filter(a_HGDP, (a > (-0.5)) & (a<0.5)) %>% select(ID) %>% pull()
HGDP_nobiasedID <- filter(HGDP_fem, ID %in% HGDP_nobiased_samples)
HGDP_clean_fem <- filter(HGDP_pcr_free, ID %in% HGDP_nobiased_samples)
```

``` r
fem_cn <- read_delim("/Volumes/Temp1/rpianezza/0.old/summary-HGDP/HGDP_cutoff_classified.tsv") %>% select(ID, country, sex, type, familyname, copynumber) %>% filter(type=="te", sex=="female", ID %in% HGDP_clean_fem$ID) %>% mutate(familyname = paste0(familyname, "_", type)) %>% group_by(familyname) %>% dplyr::summarise(afr_copynumber = mean(copynumber))
```

    ## Rows: 1394352 Columns: 12
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (9): ID, pop, sex, country, type, familyname, batch, superfamily, shared...
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
vindija_cn <- read_delim("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/data/vindija/vindija.mq10.mapstat", skip = 8, col_names = c("type", "familyname", "length", "reads", "copynumber")) %>% select(type, familyname, copynumber) %>% filter(type=="te") %>% mutate(familyname = paste0(familyname, "_", type)) %>% select(-type)
```

    ## Rows: 1703 Columns: 5
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (2): type, familyname
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
altai_cn <- read_delim("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/data/denisova-nh/denisova-nh.mq10.mapstat", skip = 8, col_names = c("type", "familyname", "length", "reads", "copynumber")) %>% select(type, familyname, copynumber) %>% filter(type=="te") %>% mutate(familyname = paste0(familyname, "_", type)) %>% select(-type)
```

    ## Rows: 1703 Columns: 5
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (2): type, familyname
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
denisova_cn <- read_delim("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/data/denisova/denisova.mq10.mapstat", skip = 8, col_names = c("type", "familyname", "length", "reads", "copynumber")) %>% select(type, familyname, copynumber) %>% filter(type=="te") %>% mutate(familyname = paste0(familyname, "_", type)) %>% select(-type)
```

    ## Rows: 1703 Columns: 5
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (2): type, familyname
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
#write_tsv(fem_cn, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/females_copynumber")
#write_tsv(vindija_cn, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/vindija_copynumber")
#write_tsv(altai_cn, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/altai_copynumber")
#write_tsv(denisova_cn, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/denisova_copynumber")
```

``` r
normalize_cn <- function(sync_matrix, total_diff_matrix, titlee){
  
  my_colors <- c("#F8766D", "#C49A00", "#53B400", "#00C094", "#00B6EB", "#A58AFF", "#FB61D7")
  
  cn <- total_diff_matrix %>% select(familyname, position, afr_copynumber, total_diff) %>% mutate(single_diff = total_diff/afr_copynumber) %>% select(-total_diff)
  merged <- inner_join(cn, sync_matrix, by=c("familyname","position")) %>% rename(copynumber = afr_copynumber)
  estimates <- merged %>% mutate_at(vars(5:ncol(merged)), ~ ((single_diff - .)/2)*copynumber ) %>% mutate_all(~ ifelse(. < 0, 0, .)) %>% select(-copynumber, -single_diff) 
  summary <- estimates %>% summarise(across(where(is.numeric) & !contains("position"), ~ sum(., na.rm = TRUE))) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1") %>% as_tibble() %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP_fem, by="ID") %>% distinct()
  
plot <- summary %>% 
  group_by(pop, country) %>% 
  summarise(mean_intro = mean(introgressed), 
            sd_intro = sd(introgressed)) %>% 
  ggplot(aes(x = reorder(pop, -mean_intro), y = mean_intro, fill = country)) +
  geom_col() +
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro),
                width = 0.2, position = position_dodge(0.9)) +
  scale_fill_manual(values = setNames(my_colors, c("Africa", "America", "Central_South_Asia", "East_Asia", "Europe", "Middle_East", "Oceania"))) +
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, titlee)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
        legend.position = "bottom", 
        plot.title = element_text(hjust = 0.5)) #+ 
  #ylim(0, NA)

plot
}
```

## 10k variants

### Cutoff = 10.000 copies

``` r
te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10.000/vindija-cutoff10k-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10.000/altai-cutoff10k-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10.000/denisova-cutoff10k-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10.000/vindija-cutoff10k-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10.000/altai-cutoff10k-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10.000/denisova-cutoff10k-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-6-1.png)<!-- -->

``` r
normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-6-2.png)<!-- -->

``` r
normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-6-3.png)<!-- -->

### Cutoff = 1.000 copies

``` r
te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.000/vindija-cutoff1k")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.000/altai-cutoff1k")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.000/denisova-cutoff1k")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.000/vindija-cutoff1k_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.000/altai-cutoff1k_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.000/denisova-cutoff1k_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq (CN < 1.000)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-9-1.png)<!-- -->

``` r
normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq (CN < 1.000)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-9-2.png)<!-- -->

``` r
normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq (CN < 1.000)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-9-3.png)<!-- -->

### Cutoff = 100 copies

``` r
te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-100/vindija-cutoff100-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-100/altai-cutoff100-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-100/denisova-cutoff100-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-100/vindija-cutoff100-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-100/altai-cutoff100-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-100/denisova-cutoff100-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq (CN < 100)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-12-1.png)<!-- -->

``` r
normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq (CN < 100)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-12-2.png)<!-- -->

``` r
normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq (CN < 100)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-12-3.png)<!-- -->

### Cutoff = 10 copies

``` r
te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10/vindija-cutoff10-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10/altai-cutoff10-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10/denisova-cutoff10-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10/vindija-cutoff10-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10/altai-cutoff10-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-10/denisova-cutoff10-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq (CN < 10)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-15-1.png)<!-- -->

``` r
normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq (CN < 10)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-15-2.png)<!-- -->

``` r
normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq (CN < 10)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-15-3.png)<!-- -->

### Cutoff = 2 copies

``` r
te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-2/vindija-cutoff2-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-2/altai-cutoff2-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-2/denisova-cutoff2-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-2/vindija-cutoff2-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-2/altai-cutoff2-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-2/denisova-cutoff2-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq (CN < 2)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-18-1.png)<!-- -->

``` r
normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq (CN < 2)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-18-2.png)<!-- -->

``` r
normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq (CN < 2)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-18-3.png)<!-- -->

### Cutoff = 1 copy

``` r
te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.2/vindija-cutoff1.2-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.2/altai-cutoff1.2-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.2/denisova-cutoff1.2-10kSNPs")
```

    ## Rows: 10000 Columns: 830
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.2/vindija-cutoff1.2-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.2/altai-cutoff1.2-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/cutoff-1.2/denisova-cutoff1.2-10kSNPs_afr-arch")
```

    ## Rows: 10000 Columns: 17
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (1): familyname
    ## dbl (16): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
#(te_denisova <- inner_join(te_denisova_fem, te_diff_denisova_fem, by=c("familyname", "position")) %>% filter(total_diff>1) %>% relocate(total_diff) %>% arrange(desc(total_diff)))

#(intro_denisova <- te_denisova %>% select(-familyname, -position, -total_diff) %>% summarise_all(~sum(. <1, na.rm=TRUE)) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1") %>% as_tibble() %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP, by="ID"))

#(te_d_plot <- intro_denisova %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(introgressed), sd_intro = sd(introgressed)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
  #geom_col()+
  #geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro), width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  #labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```

``` r
normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq (CN < 1)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-22-1.png)<!-- -->

``` r
normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq (CN < 1)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-22-2.png)<!-- -->

``` r
normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq (CN < 1)")
```

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.

![](02.archaic-SNPs_files/figure-gfm/unnamed-chunk-22-3.png)<!-- -->

## 50k variants

### Cutoff = 10.000 copies

``` r
#te_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/all/50kvindija.normalized")
#te_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/all/50kaltai.normalized")
#te_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/all/50kdenisova.normalized")
```

``` r
#te_diff_vindija_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/all/50kvindija.normalized_afr-arch")
#te_diff_altai_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/all/50kaltai.normalized_afr-arch")
#te_diff_denisova_fem <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/te/all/50kdenisova.normalized_afr-arch")
```

``` r
#normalize_cn(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq")
#normalize_cn(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq")
#normalize_cn(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisovans found in RepSeq")
```

## Test

``` r
#normalize_cn_2 <- function(sync_matrix, total_diff_matrix, titlee){
  
 # cn <- total_diff_matrix %>% select(familyname, position, afr_copynumber, total_diff) %>% mutate(single_diff = total_diff/afr_copynumber) %>% select(-total_diff)
  
  #merged <- inner_join(cn, sync_matrix, by=c("familyname","position")) %>% rename(copynumber = afr_copynumber)
  #estimates <- merged %>% mutate_at(vars(5:ncol(merged)), ~ ifelse(abs(single_diff - .) < (0.1*single_diff), 0, ((single_diff - .)/2)*copynumber)) %>% select(-copynumber, -single_diff) #%>% mutate_all(~ ifelse(. < 0, 0, .)) 
  
#  summary <- estimates %>% summarise(across(where(is.numeric) & !contains("position"), ~ sum(., na.rm = TRUE))) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1") %>% as_tibble() %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP_fem, by="ID") %>% distinct()
  
  #plot <- summary %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(introgressed)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
  #geom_col()+
  #labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, titlee)) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5))
#plot
#}

#normalize_cn_2(te_vindija_fem, te_diff_vindija_fem, "Introgressed variants from Vindija Neandertal found in RepSeq")
#normalize_cn_2(te_altai_fem, te_diff_altai_fem, "Introgressed variants from Altai Neandertal found in RepSeq")
#normalize_cn_2(te_denisova_fem, te_diff_denisova_fem, "Introgressed variants from Denisova found in RepSeq")
```
