---
title: "Archaic humans - SNPs found in NH and DH"
output: rmarkdown::github_document
editor_options: 
  markdown: 
    wrap: sentence
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '/Users/rpianezza/TE/')
```

Set the environment and creating metadata file.
```{r}
library(tidyverse)
library(umap)

HGDP <- read_delim("/Volumes/Temp1/rpianezza/0.old/summary-HGDP/HGDP_cutoff_classified.tsv") %>% select(ID, sex, country, pop) %>% distinct()

archaic <- tibble(
  ID = c("Vindija", "Altai", "Denisovan"),
  pop = c("Vindija", "Altai", "Denisova"),
  country = c("Neandertal", "Neandertal", "Denisovan"),
  sex = c("female", "female", "female")
)

HGDP_arch <- bind_rows(HGDP, archaic)

coordinates <- read_tsv("/Users/rpianezza/TE/summary-HGDP/HGDP_populationcoordinates.txt", col_names = c("pop", "region", "latitude", "longitude")) %>% select(pop, latitude, longitude)
```

```{r}
afr <- HGDP %>% filter(country=="Africa", sex=="female") %>% select(ID)
#write_tsv(afr, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/ID-Afr-females")

fem <- HGDP %>% filter(sex=="female") %>% select(ID)
#write_tsv(fem, "/Volumes/Temp1/rpianezza/PCA-SNPs-all-analysis/matrixes/females/females.IDs")

mal <- HGDP %>% filter(sex=="male") %>% select(ID)
#write_tsv(mal, "/Volumes/Temp1/rpianezza/PCA-SNPs-all-analysis/matrixes/females/males.IDs")
```


## SCGs

Reading the matrixes for the archaic variants in SCGs in the HGDP. 1 file for each archaic genome.
```{r scg-matrixes}
scg_vindija_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/vindija1k")
scg_altai_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/altai1k")
scg_denisova_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/denisova1k")
```

Read the files containing info about the variants called in the archaic genome. Is important the "total_diff" column, which is an indicator of the distance between the allele frequencies in the Africans and in the archaic genome.

Is comprised between 0 (same allele frequencies for that position) and 2 (complete non-overlapping allele frequencies for that position).

```{r scg-variants}
scg_diff_vindija <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/vindija1k_afr-arch") %>% select(familyname, position, total_diff)
scg_diff_altai <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/altai1k_afr-arch") %>% select(familyname, position, total_diff)
scg_diff_denisova <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/denisova1k_afr-arch") %>% select(familyname, position, total_diff)
```

Merge the two files read above for each of the three genome and select only the variance with a **total_diff** > a certain threshold. In this case, I use 1.75. This means that, for single copy genes, both the alleles must be different between archaic and africans to keep the variant in this code. Ideally, this will result in a total_diff = 2, but due to sequencing error and aDNA bias I use a lower threshold (1.97).

```{r scg-selected-variants}
scg_vindija <- inner_join(scg_vindija_all, scg_diff_vindija, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
scg_altai <- inner_join(scg_altai_all, scg_diff_altai, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
scg_denisova <- inner_join(scg_denisova_all, scg_diff_denisova, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
```

With the selected variants, I then look for potentially introgressed variants in all the HGDP samples: the variants which are very different between archaic and africans (total_diff very high) but that in a genome are not that different. I use a threshold of **total_diff** of 1.25 to select those variant which are supposed to be introgressed in one copy in a genome (ideally 1).

From the info above, I produce these plots averaging the number of introgressed variants for each population.

```{r}
IDs <- colnames(scg_vindija)[4:ncol(scg_vindija)]

complete_introgression <- tibble()
colnames(complete_introgression) <- c("ID", "archaic", "neandertal", "denisovan")
```


```{r}
for (id in IDs) {
  
names_v <- scg_vindija %>% select(familyname, position, id) %>% rename(sample_ID=id) %>% filter(sample_ID<1.25, na.rm=TRUE) %>% mutate(SNP = paste0(familyname, "_", position)) %>% select(SNP)
names_a <- scg_altai %>% select(familyname, position, id) %>% rename(sample_ID=id) %>% filter(sample_ID<1.25, na.rm=TRUE) %>% mutate(SNP = paste0(familyname, "_", position)) %>% select(SNP)

d_SNPs <- scg_denisova %>% select(familyname, position, id) %>% rename(sample_ID=id) %>% filter(sample_ID<1.25, na.rm=TRUE) %>% mutate(SNP = paste0(familyname, "_", position)) %>% select(SNP)
n_SNPs <- names_v %>% bind_rows(names_a) %>% distinct()

archaic_SNPs <- bind_rows(d_SNPs, n_SNPs)  %>% distinct() %>% summarise(archaic=n())
denisovan_SNPs <- d_SNPs %>% filter(!(SNP %in% n_SNPs$SNP)) %>% summarise(denisovan=n())
neandertal_SNPs <- n_SNPs %>% filter(!(SNP %in% d_SNPs$SNP)) %>% summarise(neandertal=n())

sample <- bind_cols(archaic_SNPs, neandertal_SNPs, denisovan_SNPs) %>% mutate("ID" = id)

complete_introgression <- bind_rows(complete_introgression, sample)
}
```

```{r}
data_introgression <- complete_introgression %>% separate(ID, into=c("ID","pop"), sep="\\-") %>% select(-pop) %>% inner_join(HGDP, by="ID")
```

```{r}
(archaic_introgression <- data_introgression %>% select(ID, pop, country, archaic) %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(archaic), sd_intro = sd(archaic)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
 geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro),         width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Archaic variants (SCGs)")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))
```


```{r}
neandertal_introgression <- data_introgression %>% select(ID, pop, country, neandertal)
 
(plot <- neandertal_introgression %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(neandertal), sd_intro = sd(neandertal)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
 geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro),         width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Neandertal (SCGs)")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))


full <- inner_join(neandertal_introgression, coordinates, by="pop") %>% group_by(pop, longitude, latitude) %>% dplyr::summarise(introgressed = mean(neandertal), count = n())
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(data = full, aes(x=longitude, y=latitude, color = introgressed, size = count)) + geom_errorbar() + scale_colour_gradient(low = "green", high = "red") + theme(legend.position="top") + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Neandertal (SCGs)") + labs(colour = "Introgressed variants", size = "Samples")
```

```{r}
denisovan_introgression <- data_introgression %>% select(ID, pop, country, denisovan)
 
(plot <-denisovan_introgression %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(denisovan), sd_intro = sd(denisovan)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
 geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro),         width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, "Denisovan (SCGs)")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5)))


full <- inner_join(denisovan_introgression, coordinates, by="pop") %>% group_by(pop, longitude, latitude) %>% dplyr::summarise(introgressed = mean(denisovan), count = n())
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(data = full, aes(x=longitude, y=latitude, color = introgressed, size = count)) + geom_errorbar() + scale_colour_gradient(low = "green", high = "red") + theme(legend.position="top") + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Denisovan (SCGs)") + labs(colour = "Introgressed variants", size = "Samples")
```

