HGDP - Analyzing the TE abundance in the two sexes
================

``` r
library(tidyverse)
```

    ## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
    ## ✔ dplyr     1.1.1     ✔ readr     2.1.4
    ## ✔ forcats   1.0.0     ✔ stringr   1.5.0
    ## ✔ ggplot2   3.4.2     ✔ tibble    3.2.1
    ## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
    ## ✔ purrr     1.0.1     
    ## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()
    ## ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

``` r
library(ggpubr)

theme_set(theme_bw())

HGDP <- read_delim("/Volumes/Temp1/rpianezza/0.old/summary-HGDP/HGDP_cutoff_classified.tsv")
```

    ## Rows: 1394352 Columns: 12
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (9): ID, pop, sex, country, type, familyname, batch, superfamily, shared...
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
a_HGDP <- read_tsv("/Volumes/Temp1/rpianezza/PCA-copynumber-all-analysis/a_HGDP.tsv")
```

    ## Rows: 828 Columns: 2
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (1): ID
    ## dbl (1): a
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
HGDP_nobiased_samples <- filter(a_HGDP, (a > (-0.5)) & (a<0.5)) %>% select(ID) %>% pull()
HGDP_clean <- filter(HGDP, ID %in% HGDP_nobiased_samples)

DNA_names <- c("Crypton", "hAT", "Helitron", "Kolobok", "Mariner/Tc1", "Merlin", "MuDR", "piggyBac", "DNA transposon")
LINE_names <- c("L1", "CR1", "L2", "Crack", "RTE", "RTEX", "R4", "Vingi", "Tx1", "Penelope")
SINE_names <- c("SINE1/7SL", "SINE2/tRNA", "SINE3/5S", "SINE")
LTR_names <- c("ERV1", "ERV2", "ERV3", "Gypsy", "Endogenous Retrovirus", "LTR Retrotransposon", "Long terminal repeat", "Non-LTR Retrotransposon")
satellites_names <- c("Satellite", "satellite", "SAT")

classification <- HGDP_clean %>% mutate(class = case_when(superfamily %in% DNA_names ~ "DNA", superfamily %in% LINE_names ~ "LINE", superfamily %in% SINE_names ~ "SINE", superfamily %in% LTR_names ~ "LTR", superfamily %in% satellites_names ~ "satellite"))
```

# Sex differences

Male genome length = 6,045,293,052 bp Female genome length =
5,948,152,252 bp

``` r
f = 6045293052
m = 5948152252

diff_mf = f-m
diff_perc = (diff_mf/mean(f, m))*100

diff_perc
```

    ## [1] 1.606883

We expect a slight increase in females copynumber due to the bigger
genome they have: **1.6%** bigger than males (Y chromosome is shorter
than the X).

We chose a conservative threshold of **3.2%** to state that a TE
copynumber is significantly different between the sexes despite the
genome length difference.

``` r
cn <- classification %>% filter(type=="te") %>% group_by(familyname, sex, class, superfamily) %>% summarise(mean = mean(copynumber)) %>% filter(mean>1) #%>% filter(mean < 1000)
```

    ## `summarise()` has grouped output by 'familyname', 'sex', 'class'. You can
    ## override using the `.groups` argument.

``` r
(new_tibble <- tidyr::spread(cn, key = sex, value = mean))
```

    ## # A tibble: 714 × 5
    ## # Groups:   familyname, class [714]
    ##    familyname class     superfamily    female      male
    ##    <chr>      <chr>     <chr>           <dbl>     <dbl>
    ##  1 6kbHsap    satellite satellite      299.      302.  
    ##  2 ALR        satellite SAT          31525.    31812.  
    ##  3 ALR_       satellite SAT          77235.    76912.  
    ##  4 ALR1       satellite SAT          70870.    71586.  
    ##  5 ALR2       satellite SAT           2087.     2057.  
    ##  6 ALRa       satellite SAT            371.      361.  
    ##  7 ALRa_      satellite SAT           1776.     1750.  
    ##  8 ALRb       satellite SAT          28610.    28660.  
    ##  9 ALU        SINE      SINE1/7SL   196091.   192089.  
    ## 10 BLACKJACK  DNA       hAT              2.00      1.92
    ## # ℹ 704 more rows

``` r
colnames(new_tibble) <- c("familyname", "class", "superfamily", "female_cn", "male_cn")

(different <- new_tibble %>% mutate(mean_cn = (female_cn+male_cn)/2, diff = abs(female_cn-male_cn)) %>% filter(diff > ((mean_cn/100)*3.2), diff>1) %>% arrange(desc(diff)))
```

    ## # A tibble: 128 × 7
    ## # Groups:   familyname, class [128]
    ##    familyname class     superfamily female_cn male_cn mean_cn   diff
    ##    <chr>      <chr>     <chr>           <dbl>   <dbl>   <dbl>  <dbl>
    ##  1 HSATI      satellite SAT              341.   2122.   1232. 1781. 
    ##  2 HSATII     satellite SAT            28262.  27132.  27697. 1130. 
    ##  3 THE1_I     LTR       ERV3            4436.   4273.   4355.  164. 
    ##  4 L1PA4      LINE      L1              4347.   4195.   4271.  152. 
    ##  5 L1PA7_5    LINE      L1              1945.   1853.   1899.   91.4
    ##  6 L1PREC1    LINE      L1              2216.   2129.   2172.   86.6
    ##  7 L1PA16     LINE      L1              2205.   2131.   2168.   73.9
    ##  8 L1PB2c     LINE      L1              1487.   1424.   1456.   62.3
    ##  9 L1PA7      LINE      L1              1806.   1747.   1777.   58.8
    ## 10 L1PB1      LINE      L1              1303.   1251.   1277.   52.7
    ## # ℹ 118 more rows

``` r
(more_in_males <- different %>% filter(male_cn>female_cn))
```

    ## # A tibble: 8 × 7
    ## # Groups:   familyname, class [8]
    ##   familyname class     superfamily           female_cn male_cn mean_cn    diff
    ##   <chr>      <chr>     <chr>                     <dbl>   <dbl>   <dbl>   <dbl>
    ## 1 HSATI      satellite SAT                      341.    2122.  1232.   1781.  
    ## 2 LTR14C     LTR       ERV2                      67.1     76.1   71.6     8.93
    ## 3 HERV9      LTR       ERV1                     134.     142.   138.      7.35
    ## 4 LTR22      LTR       ERV2                      47.5     53.5   50.5     6.07
    ## 5 LTR6A      LTR       ERV1                      68.7     71.3   70.0     2.67
    ## 6 HERV-K14CI LTR       ERV2                       8.63    10.9    9.76    2.28
    ## 7 LTR22B     LTR       ERV2                      41.6     43.6   42.6     2.00
    ## 8 HERV30I    LTR       Endogenous Retrovirus      9.77    10.9   10.3     1.09

``` r
(more_in_females <- different %>% filter(male_cn<female_cn))
```

    ## # A tibble: 120 × 7
    ## # Groups:   familyname, class [120]
    ##    familyname class     superfamily female_cn male_cn mean_cn   diff
    ##    <chr>      <chr>     <chr>           <dbl>   <dbl>   <dbl>  <dbl>
    ##  1 HSATII     satellite SAT            28262.  27132.  27697. 1130. 
    ##  2 THE1_I     LTR       ERV3            4436.   4273.   4355.  164. 
    ##  3 L1PA4      LINE      L1              4347.   4195.   4271.  152. 
    ##  4 L1PA7_5    LINE      L1              1945.   1853.   1899.   91.4
    ##  5 L1PREC1    LINE      L1              2216.   2129.   2172.   86.6
    ##  6 L1PA16     LINE      L1              2205.   2131.   2168.   73.9
    ##  7 L1PB2c     LINE      L1              1487.   1424.   1456.   62.3
    ##  8 L1PA7      LINE      L1              1806.   1747.   1777.   58.8
    ##  9 L1PB1      LINE      L1              1303.   1251.   1277.   52.7
    ## 10 L1PREC2    LINE      L1              1007.    956.    982.   51.2
    ## # ℹ 110 more rows

``` r
m_count <- more_in_males %>% ungroup() %>% summarise(count = n()) %>% pull()
f_count <- more_in_females %>% ungroup() %>% summarise(count = n()) %>% pull()
tot_count <- classification %>% filter(type=="te") %>% select(familyname) %>% distinct() %>% summarise(count = n()-(m_count+f_count)) %>% pull()
names <- c("More in males", "More in females", "No difference")

pie_tibble <- tibble(m_count, f_count, tot_count) %>% t() %>% as_tibble() %>% bind_cols(names) %>% rename("count" = "V1", "condition" = "...2") %>% relocate(condition, before = NULL)
```

    ## Warning: The `x` argument of `as_tibble.matrix()` must have unique column names if
    ## `.name_repair` is omitted as of tibble 2.0.0.
    ## ℹ Using compatibility `.name_repair`.
    ## This warning is displayed once every 8 hours.
    ## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
    ## generated.

    ## New names:
    ## • `` -> `...2`

``` r
pie_tibble <- pie_tibble %>% mutate(percent = count/sum(pie_tibble$count)*100)

(pie <- ggplot(pie_tibble, aes(x="", y=count, fill=condition))+
  geom_col(width = 1) + labs(fill='') +
  coord_polar(theta="y", start = pi/2)+
  theme_void()+
  geom_text(aes(label = paste0(round(percent, 1), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 4))+
  theme(legend.position = "left")+
  scale_fill_manual(values=c("#FFFF33", "#FF9900", "#999999"))
```

![](00_files/figure-gfm/unnamed-chunk-4-1.png)<!-- -->

``` r
females <- more_in_females %>% group_by(class) %>% summarise(count = n()) 

females <- females %>% mutate(percent = count/sum(females$count)*100)

sum_satellite_na <- females %>%
  filter(class %in% c("satellite", NA)) %>%
  summarise(count = sum(count),
            percent = sum(percent, na.rm = TRUE)) %>% mutate(class = "Other")

(females_mod <- females %>% filter(!(class %in% c("satellite", NA))) %>% bind_rows(sum_satellite_na))
```

    ## # A tibble: 4 × 3
    ##   class count percent
    ##   <chr> <int>   <dbl>
    ## 1 DNA       5    4.17
    ## 2 LINE     49   40.8 
    ## 3 LTR      63   52.5 
    ## 4 Other     3    2.5

``` r
x_values <- ifelse(females_mod$class == "Other", 1.3, 1.3)

(f_pie <- ggplot(females_mod, aes(x="", y=count, fill=class))+
  geom_col(width = 1) + labs(fill='') +
  coord_polar(theta="y")+
  theme_void()+
  geom_text(aes(x = x_values, label = paste0(round(percent, 1), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 3)+
  scale_color_manual(values=c("#F8766D", "#7CAE00", "#00BFC4", "#C77CFF")))
```

![](00_files/figure-gfm/unnamed-chunk-5-1.png)<!-- -->

``` r
males <- more_in_males %>% group_by(class) %>% summarise(count = n()) 

males <- males %>% mutate(percent = count/sum(males$count)*100)

sum_satellite_na <- males %>%
  filter(class %in% c("satellite", NA)) %>%
  summarise(count = sum(count),
            percent = sum(percent, na.rm = TRUE)) %>% mutate(class = "Other")

(males_mod <- males %>% filter(!(class %in% c("satellite", NA))) %>% bind_rows(sum_satellite_na))
```

    ## # A tibble: 2 × 3
    ##   class count percent
    ##   <chr> <int>   <dbl>
    ## 1 LTR       7    87.5
    ## 2 Other     1    12.5

``` r
x_values <- ifelse(males_mod$class == "Other", 1.3, 1.3)

(m_pie <- ggplot(males_mod, aes(x="", y=count, fill=class))+
  geom_col(width = 1) + labs(fill='') +
  coord_polar(theta="y")+
  theme_void()+
  geom_text(aes(x = x_values, label = paste0(round(percent, 1), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 3)+
  scale_fill_manual(values=c("#00BFC4", "#C77CFF")))
```

![](00_files/figure-gfm/unnamed-chunk-5-2.png)<!-- -->

# CV

``` r
cn <- classification %>% filter(type=="te") %>% group_by(familyname, class, superfamily) %>% summarise(mean = mean(copynumber), var = var(copynumber), cv = var/mean) %>% arrange(desc(cv)) %>% filter(!(class %in% c("satellite", NA)), mean>1) %>% filter(cv > 1.58)
```

    ## `summarise()` has grouped output by 'familyname', 'class'. You can override
    ## using the `.groups` argument.

``` r
outliers <- HGDP_clean %>% filter(type=="te", familyname %in% cn$familyname) %>% inner_join(cn, by="familyname")
outliers <- outliers[order(outliers$copynumber,decreasing=T),]
outliers$familyname<-factor(outliers$familyname,levels=unique(outliers$familyname))

(box <- ggplot(outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylim(0,7000)+
 scale_color_manual(values=c("#F8766D", "#7CAE00", "#00BFC4", "#C77CFF")))
```

    ## Warning: Removed 660 rows containing non-finite values (`stat_boxplot()`).

![](00_files/figure-gfm/unnamed-chunk-6-1.png)<!-- -->

``` r
(f_cn <- classification %>% filter(type=="te", sex=="female") %>% group_by(familyname, class, superfamily) %>% summarise(mean = mean(copynumber), var = var(copynumber), cv = var/mean) %>% arrange(desc(cv)) %>% filter(!(class %in% c("satellite", NA)), mean>1) %>% filter(cv > 1.62))
```

    ## `summarise()` has grouped output by 'familyname', 'class'. You can override
    ## using the `.groups` argument.

    ## # A tibble: 30 × 6
    ## # Groups:   familyname, class [30]
    ##    familyname class superfamily    mean       var     cv
    ##    <chr>      <chr> <chr>         <dbl>     <dbl>  <dbl>
    ##  1 ALU        SINE  SINE1/7SL   196091. 29865070. 152.  
    ##  2 THE1B      LTR   ERV3          6287.    61307.   9.75
    ##  3 L1PA4      LINE  L1            4347.    41456.   9.54
    ##  4 L1         LINE  L1            3661.    27735.   7.58
    ##  5 L1PREC1    LINE  L1            2216.    13149.   5.93
    ##  6 THE1C      LTR   ERV3          3423.    18428.   5.38
    ##  7 L1PA16     LINE  L1            2205.    11478.   5.20
    ##  8 L1PA6      LINE  L1            2120.    10417.   4.91
    ##  9 THE1A      LTR   ERV3          2792.    13475.   4.83
    ## 10 THE1D      LTR   ERV3          2753.    13041.   4.74
    ## # ℹ 20 more rows

``` r
f_outliers <- HGDP_clean %>% filter(type=="te", sex=="female", familyname %in% f_cn$familyname) %>% inner_join(f_cn, by="familyname")
f_outliers <- f_outliers[order(f_outliers$copynumber,decreasing=T),]
f_outliers$familyname<-factor(f_outliers$familyname,levels=unique(f_outliers$familyname))

(box <- ggplot(f_outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylim(0,7000)+
 scale_color_manual(values=c("#F8766D", "#7CAE00", "#00BFC4", "#C77CFF")))
```

    ## Warning: Removed 254 rows containing non-finite values (`stat_boxplot()`).

![](00_files/figure-gfm/unnamed-chunk-7-1.png)<!-- -->

``` r
(m_cn <- classification %>% filter(type=="te", sex=="male") %>% group_by(familyname, class, superfamily) %>% summarise(mean = mean(copynumber), var = var(copynumber), cv = var/mean) %>% arrange(desc(cv)) %>% filter(!(class %in% c("satellite", NA)), mean>1) %>% filter(cv > 1.05))
```

    ## `summarise()` has grouped output by 'familyname', 'class'. You can override
    ## using the `.groups` argument.

    ## # A tibble: 30 × 6
    ## # Groups:   familyname, class [30]
    ##    familyname class superfamily    mean       var     cv
    ##    <chr>      <chr> <chr>         <dbl>     <dbl>  <dbl>
    ##  1 ALU        SINE  SINE1/7SL   192089. 26394978. 137.  
    ##  2 L1PA4      LINE  L1            4195.    26884.   6.41
    ##  3 THE1B      LTR   ERV3          6134.    37824.   6.17
    ##  4 L1         LINE  L1            3551.    17032.   4.80
    ##  5 L1PREC1    LINE  L1            2129.     8038.   3.78
    ##  6 THE1_I     LTR   ERV3          4273.    15185.   3.55
    ##  7 L1PA16     LINE  L1            2131.     7356.   3.45
    ##  8 THE1C      LTR   ERV3          3330.    11242.   3.38
    ##  9 THE1D      LTR   ERV3          2676.     8149.   3.04
    ## 10 THE1A      LTR   ERV3          2733.     8302.   3.04
    ## # ℹ 20 more rows

``` r
m_outliers <- HGDP_clean %>% filter(type=="te", sex=="male", familyname %in% m_cn$familyname) %>% inner_join(m_cn, by="familyname")
m_outliers <- m_outliers[order(m_outliers$copynumber,decreasing=T),]
m_outliers$familyname<-factor(m_outliers$familyname,levels=unique(m_outliers$familyname))

(box <- ggplot(m_outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylim(0,7000)+
 scale_color_manual(values=c("#F8766D", "#7CAE00", "#00BFC4", "#C77CFF")))
```

    ## Warning: Removed 406 rows containing non-finite values (`stat_boxplot()`).

![](00_files/figure-gfm/unnamed-chunk-7-2.png)<!-- -->

``` r
(box_ALU <- ggplot(outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylim(10000,220000)+
 scale_color_manual(values=c("#D89000")))
```

    ## Warning: Removed 18480 rows containing non-finite values (`stat_boxplot()`).

![](00_files/figure-gfm/unnamed-chunk-8-1.png)<!-- -->

``` r
(m_box_ALU <- ggplot(m_outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylim(150000,220000)+
 scale_color_manual(values=c("#D89000")))
```

    ## Warning: Removed 11774 rows containing non-finite values (`stat_boxplot()`).

![](00_files/figure-gfm/unnamed-chunk-8-2.png)<!-- -->

``` r
(f_box_ALU <- ggplot(f_outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylim(150000,220000)+
 scale_color_manual(values=c("#D89000")))
```

    ## Warning: Removed 7366 rows containing non-finite values (`stat_boxplot()`).

![](00_files/figure-gfm/unnamed-chunk-8-3.png)<!-- -->

``` r
cn <- classification %>% filter(type=="te") %>% group_by(familyname, class, superfamily) %>% summarise(mean = mean(copynumber), var = var(copynumber), cv = var/mean) %>% arrange(desc(cv)) %>% filter(!(class %in% c("satellite", NA)), mean>1) %>% filter(cv > 1.58)
```

    ## `summarise()` has grouped output by 'familyname', 'class'. You can override
    ## using the `.groups` argument.

``` r
outliers <- HGDP_clean %>% filter(type=="te", familyname %in% cn$familyname) %>% inner_join(cn, by="familyname")
outliers <- outliers[order(outliers$copynumber,decreasing=T),]
outliers$familyname<-factor(outliers$familyname,levels=unique(outliers$familyname))

(box <- ggplot(outliers, aes(x=familyname, y=copynumber)) + geom_boxplot(notch=F, aes(color=class)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
 scale_color_manual(values=c("#F8766D", "#7CAE00", "#00BFC4", "#D89000")))
```

![](00_files/figure-gfm/unnamed-chunk-9-1.png)<!-- -->

``` r
library(scales)
```

    ## 
    ## Attaching package: 'scales'

    ## The following object is masked from 'package:purrr':
    ## 
    ##     discard

    ## The following object is masked from 'package:readr':
    ## 
    ##     col_factor

``` r
#extract hex color codes for a plot with three elements in ggplot2 
hex <- hue_pal()(10)

#overlay hex color codes on actual colors
show_col(hex)
```

![](00_files/figure-gfm/unnamed-chunk-10-1.png)<!-- -->
