---
title: "HGDP - Controls"
output: rmarkdown::github_document
editor_options: 
  markdown: 
    wrap: sentence
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '/Users/rpianezza/TE/')
```

Script 5. This scripts contains some controls on the data quality.
```{r}
library(tidyverse)
library(ggpubr)

(HGDPcutoff<-read_delim("/Users/rpianezza/TE/summary-HGDP/USEME_HGDP_complete_reflib6.2_mq10_batchinfo_cutoff0.01.txt",comment="#"))
names(HGDPcutoff)<-c("ID","Pop","sex","Country","type","familyname","length","reads","copynumber","batch")
```

The two following functions have been previously described in scripts 1 and 4. The first plot the distribution of the **copynumbers** of a TE divided by **continent**. The second plot the **mean copynumber** of every **population** for a particular TE on the **world map**.

These functions are slightly modified in order to add `batch` among the arguments and check for batch effect. 
```{r}
plotTEfamily <- function(data, famname, bat, binwidht, x_title, y_title, x_numbers, y_numbers){
filtered <- filter(data, familyname==famname, batch==bat)
ggplot(data = filtered, mapping = aes(x = copynumber, fill = Country)) +
  geom_histogram(binwidth = binwidht) +
  facet_wrap(~sex)+
  ggtitle(paste0(filtered$batch, ' - ',famname)) + theme(plot.title = element_text(size = 8, hjust = 0.5)) +
  {if(x_title=='n'){
  theme(axis.title.x=element_blank())}} +
  {if(y_title=='n'){
  theme(axis.title.y=element_blank())}} +
  {if(x_numbers=='n'){
  theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())}} +
{if(y_numbers=='n'){
  theme(axis.text.y=element_blank(), axis.ticks.y=element_blank())}}
}
```

```{r}
plot_map <- function(data, famname, batch){
TE <- filter(data, familyname == famname)
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(
    data = TE, aes(longitude, latitude, color = copynumber, size = count)
  ) + scale_colour_gradient(low = "green", high = "red") + theme(legend.position="top") + theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(~sex) + ggtitle(paste0(batch, ' - ',famname))}
```

## Batch effect

In our dataset, the `batch` column refers to the computer on which the pipeline was ran for each sample, thus `flo` for Florian's PC and `ro` for Robert's PC. This could introduce some biases, even though the pipeline are the same, with the exact same version for each of the used programs. To check if this is the case and if the bias is significant, I did the same plots previously described in scripts 1-2-4 but dividing the dataset for the two **batches**. I choose the TE `L1ME5` which was the most interesting.
```{r}
(ro_L1ME5_plot<-plotTEfamily(HGDPcutoff, 'L1ME5', 'ro', 1, 'y', 'y', 'y', 'y'))
(flo_L1ME5_plot<-plotTEfamily(HGDPcutoff, 'L1ME5', 'flo', 1, 'y', 'y', 'y', 'y'))
```
```{r}
coordinates <- read_tsv("/Users/rpianezza/TE/summary-HGDP/HGDP_populationcoordinates.txt", col_names = c("Pop", "region", "latitude", "longitude"))
coord <- select(coordinates, Pop, latitude, longitude)

ro_TE <- filter(HGDPcutoff, type=='te', batch=='ro')
flo_TE <- filter(HGDPcutoff, type=='te', batch=='flo')

ro_by_pop <- group_by(ro_TE, Pop, familyname, sex) %>% summarise(copynumber = mean(copynumber), count=n())
ro_data <- inner_join(x = coord, y = ro_by_pop, by = "Pop")

flo_by_pop <- group_by(flo_TE, Pop, familyname, sex) %>% summarise(copynumber = mean(copynumber), count=n())
flo_data <- inner_join(x = coord, y = flo_by_pop, by = "Pop")

(ro_L1ME5_map<-plot_map(ro_data, 'L1ME5', 'ro'))
(flo_L1ME5_map<-plot_map(flo_data, 'L1ME5', 'flo'))
```

I do not notice significant differences among the two batches, in neither of the two figures.


## Read length

Another possible bias could be introduced by different **read lengths** of the samples. To check this, I found the original dataset here <https://www.ebi.ac.uk/ena/browser/view/PRJEB6463> on the ENA website. Then I selected the columns containing the number of reads for each sample (`read_count`) and the total base number sequenced in each sample (`base_count`), and finally I downloaded the report in TSV format. Here I import the report in R as `HGDP_report` and I add the column `read_lenght`, calculated as the ratio between `base_count` and `read_count`.
```{r}
HGDP_report <- read_tsv('/Users/rpianezza/TE/ric-documentation-Rmd/other-files/filereport_read_run_PRJEB6463_tsv.txt') %>% mutate(read_length = base_count/read_count)
```

First I wanted to check if the dataset used for my data analysis was still complete respect to the original one, so I compared the number of individuals in the two datasets. It's `828` in both, confirming the completness of the used dataset.
```{r}
length(unique(HGDP_report$sample_accession)) == length(unique(HGDPcutoff$ID))
```

Then I checked for the mean of read lenghts among all the samples. Surprisingly it is not `151`, as I was expecting looking at the dataset, but a bit lower.
```{r}
(mean_rl = mean(HGDP_report$read_length))
```

Thus, I checked for the samples with `read_lenght` different from the expected `151`.
```{r}
(not151 <- filter(HGDP_report, !(read_length==151)))
```

We found 14/828 samples for which read length is not `151`, but in a range between `141` and `146`. I think that these outliers are not introducing a significant bias, since the read length is very similar to the other samples. To further investigate this, I also checked from which populations these samples come from.
```{r}
out_rl_names <- c(not151$sample_alias)
out_rl <- filter(HGDPcutoff, type == "te", ID %in% out_rl_names)
(unique(out_rl$Pop))
```

The samples are distributed among 9 different populations, confirming the non significance of these outliers on the final results.


## Coverage of the aligned reads by position

Another possible problem that may create some artefacts in the data is the sequence coverage by position: a TE copynumber may be overestimated if only a small portion of the consensus sequence has many reads aligned, while the rest of the sequence being at low coverage. To test for this, I downloaded 5/828 samples (randomly) to check the coverage for some of the most interesting TEs. The files has extension *.bed.gz* and can be downloaded from <https://sourceforge.net/p/human-te-dynamics/data/HEAD/tree/raw-data/>, divided among the folders **flo_vetgrid** and **ro_vetgrid**.
```{r}
bedfile1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/HGDP00001-Brahui.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
bedfile2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/HGDP00052-Balochi.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
bedfile3 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/HGDP00099-Hazara.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
bedfile4 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/HGDP00130-Makrani.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
bedfile5 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/HGDP00163-Sindhi.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
```

Then I wrote this function to plot the coverage for a set of TEs (`familynames`) in an individual (`data`). The other arguments are purely for visualization (to include/exclude title, scale and ticks on the two axes).
```{r}
plot_bed <- function(data, familynames, x_title, y_title, x_numbers, y_numbers){

  bed <- filter(data, familyname %in% familynames)

  ggplot(bed)+
    geom_segment(aes(x=start_pos, xend=end_pos, y=coverage, yend=coverage))+
    ylab("Coverage") + xlab("Base number") +
    ggtitle(paste0(familynames)) + theme(plot.title = element_text(size = 8, hjust = 0.5)) +
    {if(x_title=='n'){
    theme(axis.title.x=element_blank())}} +
    {if(y_title=='n'){
    theme(axis.title.y=element_blank())}} +
    {if(x_numbers=='n'){
    theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())}} +
    {if(y_numbers=='n'){
    theme(axis.text.y=element_blank(), axis.ticks.y=element_blank())}}
}
```

This second function put together the plots created with the first function in a single figure. To change the TEs investigated, change the names here in this function.
```{r}
create_figure_mix <- function(file){
  
  scg <- plot_bed(file, "chr1:916864-921016_scg", 'n', 'n', 'n', 'n')
  L2 <- plot_bed(file, "L2_te", 'n', 'n', 'n', 'n')
  L1PB1 <- plot_bed(file, "L1PB1_te", 'n', 'n', 'n', 'n')
  L1ME5 <- plot_bed(file, "L1ME5_te", 'n', 'n', 'n', 'n')
  L1PA7_5 <- plot_bed(file, "L1PA7_5_te", 'n', 'n', 'n', 'n')
  ALU <- plot_bed(file, "ALU_te", 'n', 'n', 'n', 'n')
  SVA_A <- plot_bed(file, "SVA_A_te", 'n', 'n', 'n', 'n')
  MER2 <- plot_bed(file, "MER2_te", 'n', 'n', 'n', 'n')
  MLT2A1 <- plot_bed(file, "MLT2A1_te", 'n', 'n', 'n', 'n')
  
  bed_figure <- ggarrange(scg, L2, L1PB1, L1ME5, L1PA7_5, ALU, SVA_A, MER2, MLT2A1, ncol = 3, nrow = 3, common.legend = TRUE, legend = "top", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

  bed_final <- annotate_figure(bed_figure, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), fig.lab = "")
}
```

Here I use the 2 functions to create 5 figures for 5 individuals previously downloaded for a set of chosen sequences. I choose to include:

* A **single copy gene** as a control (`chr1:916864-921016_scg`).
* An **ancient** TE, expected to be inactive in the human genome (`L2`).
* 3 **L1 TEs** previously noticed to show interesting geographical distributions (`L1PB1`, `L1ME5`, `L1PA7_5`).
* 2 **SINEs**, `ALU` and `SVA_A` known to be active humans and with interesting geographical distributions.
* A **DNA transposon** which showed an interesting geographical distribution (`MER2`)
* An **endogenous retrovirus** which showed an interesting geographical distribution (`MLT2A1`)

I removed the last 2 figures because they didn't show other interesting patterns.
```{r}
# File 1: HGDP00001-Brahui.per-base.bed.gz
(create_figure_mix(bedfile1))

# File 2: HGDP00052-Balochi.per-base.bed.gz
(create_figure_mix(bedfile2))

# File 3: HGDP00099-Hazara.per-base.bed.gz
(create_figure_mix(bedfile3))

# File 4: HGDP00130-Makrani.per-base.bed.gz
#(create_figure_mix(bedfile4))

# File 5: HGDP00163-Sindhi.per-base.bed.gz
#(create_figure_mix(bedfile5))
```

Note that each plot has is own scale. What we can say from this plot is that all the analyzed sequences show similar coverage distributions between different individuals, with the exception of `L1ME5_te`. This the shows a "**spiked**" pattern is some individuals (2, 3) and a "**dashed**" pattern in others (1).

I wonder how we can explain this patterns. To check if it's related to **sex** or to **population** or to both, I checked the coverage distribution for `L1ME5` in 6 different populations, represented by 1 individuals per sex each, for a total of 12 individuals. I choose the 6 populations across the different continents:

* `Yoruba` from **Africa**
* `French` from **Europe**
* `Maya` from **America**
* `PapuanSepik` from **Oceania**
* `Japanese` from **East Asia**
* `Kalash` from **Central Asia**
```{r}
# Yoruba
yoruba_m <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00929-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
yoruba_f <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00920-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# French
french_m <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00511-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
french_f <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00513-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# Maya
maya_m <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00856-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
maya_f <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00854-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# PapuanSepik
papuansepik_m <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00540-PapuanSepik.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
papuansepik_f <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00544-PapuanSepik.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# Japanese
japanese_m <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00747-Japanese.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
japanese_f <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00754-Japanese.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# Kalash
kalash_m <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00277-Kalash.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
kalash_f <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/by_pop/HGDP00274-Kalash.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
```

```{r}
m_L1ME5_yoruba <- plot_bed(yoruba_m, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba")
m_L1ME5_french <- plot_bed(french_m, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French")
m_L1ME5_maya <- plot_bed(maya_m, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya")
m_L1ME5_papuan <- plot_bed(papuansepik_m, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("PapuanSepik")
m_L1ME5_japanese <- plot_bed(japanese_m, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Japanese")
m_L1ME5_kalash <- plot_bed(kalash_m, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Kalash")

m_pop_figure <- ggarrange(m_L1ME5_yoruba, m_L1ME5_french, m_L1ME5_maya, m_L1ME5_papuan, m_L1ME5_japanese, m_L1ME5_kalash, ncol = 3, nrow = 2, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(m_pop_final <- annotate_figure(m_pop_figure, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1ME5 - Males", color = "black"), fig.lab = ""))
  

f_L1ME5_yoruba <- plot_bed(yoruba_f, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba")
f_L1ME5_french <- plot_bed(french_f, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French")
f_L1ME5_maya <- plot_bed(maya_f, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya")
f_L1ME5_papuan <- plot_bed(papuansepik_f, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("PapuanSepik")
f_L1ME5_japanese <- plot_bed(japanese_f, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Japanese")
f_L1ME5_kalash <- plot_bed(kalash_f, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Kalash")

f_pop_figure <- ggarrange(f_L1ME5_yoruba, f_L1ME5_french, f_L1ME5_maya, f_L1ME5_papuan, f_L1ME5_japanese, f_L1ME5_kalash, ncol = 3, nrow = 2, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(f_pop_final <- annotate_figure(f_pop_figure, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1ME5 - Females", color = "black"), fig.lab = ""))
```

The **dashed** pattern is private to **Africans (m/f)**, **Oceanians (m/f)** and to **American males**, while **American females** show the **spiked** pattern, together with all the other samples.

I think it's interesting: despite that this weird coverage distribution is probably invalidating some of our results on `L1ME5`, what we see here further suggest genetic similarities among **Africa** and **Oceania**. About the **Americans** this plots suggest that they in some way inherited some variants on the **Y chromosome** private to **Africans** and **Oceanians**, while their **X chromosome** is more close to **Eurasians**.

But are this patterns consistent among all the individuals? I checked this in 3 males and 3 females from these 3 populations: `Yoruba`, `French` and `Maya` (only 2 males samples for Maya were available).
```{r}
# Yoruba: Males (HGDP00929, HGDP00937, HGDP00944) - Females (HGDP00920, HGDP00933, HGDP00939)
# French: Males (HGDP00511, HGDP00528, HGDP00538) - Females (HGDP00513, HGDP00527, HGDP00539)
# Maya: Males (HGDP00856, HGDP00877) - Females (HGDP00854, HGDP00865, HGDP00876)

# Yoruba
yoruba_m1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/yoruba/HGDP00929-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
yoruba_m2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/yoruba/HGDP00937-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
yoruba_m3 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/yoruba/HGDP00944-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
yoruba_f1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/yoruba/HGDP00920-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
yoruba_f2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/yoruba/HGDP00933-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
yoruba_f3 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/yoruba/HGDP00939-Yoruba.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# French
french_m1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/french/HGDP00511-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
french_m2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/french/HGDP00528-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
french_m3 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/french/HGDP00538-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
french_f1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/french/HGDP00513-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
french_f2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/french/HGDP00527-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
french_f3 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/french/HGDP00539-French.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
  
# Maya
maya_m1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/maya/HGDP00856-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
maya_m2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/maya/HGDP00877-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
maya_f1 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/maya/HGDP00854-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
maya_f2 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/maya/HGDP00865-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
maya_f3 <- read_tsv("/Users/rpianezza/TE/coverage_validation/data/full_pops/maya/HGDP00876-Maya.per-base.bed.gz", col_names = c("familyname", "start_pos", "end_pos", "coverage"))
```

```{r}
m1_L1ME5_yoruba <- plot_bed(yoruba_m1, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
m2_L1ME5_yoruba <- plot_bed(yoruba_m2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
m3_L1ME5_yoruba <- plot_bed(yoruba_m3, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
m1_L1ME5_french <- plot_bed(french_m1, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French 1")
m2_L1ME5_french <- plot_bed(french_m2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French 2")
m3_L1ME5_french <- plot_bed(french_m3, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French 3")
m1_L1ME5_maya <- plot_bed(maya_m1, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
m2_L1ME5_maya <- plot_bed(maya_m2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 2")

m_pop3_figure <- ggarrange(m1_L1ME5_yoruba, m2_L1ME5_yoruba, m3_L1ME5_yoruba, m1_L1ME5_french, m2_L1ME5_french, m3_L1ME5_french, m1_L1ME5_maya,  m2_L1ME5_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(m_pop3_final <- annotate_figure(m_pop3_figure, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1ME5 - Males", color = "black"), fig.lab = ""))
  

f1_L1ME5_yoruba <- plot_bed(yoruba_f1, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
f2_L1ME5_yoruba <- plot_bed(yoruba_f2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
f3_L1ME5_yoruba <- plot_bed(yoruba_f3, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
f1_L1ME5_french <- plot_bed(french_f1, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French 1")
f2_L1ME5_french <- plot_bed(french_f2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French 2")
f3_L1ME5_french <- plot_bed(french_f3, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("French 3")
f1_L1ME5_maya <- plot_bed(maya_f1, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
f2_L1ME5_maya <- plot_bed(maya_f2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 2")
f3_L1ME5_maya <- plot_bed(maya_f2, "L1ME5_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 3")

f_pop3_figure <- ggarrange(f1_L1ME5_yoruba, f2_L1ME5_yoruba, f3_L1ME5_yoruba, f1_L1ME5_french, f2_L1ME5_french, f3_L1ME5_french, f1_L1ME5_maya,  f2_L1ME5_maya, f3_L1ME5_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(f_pop3_final <- annotate_figure(f_pop3_figure, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1ME5 - Females", color = "black"), fig.lab = ""))
  
```

```{r}
m1_L1PB1_yoruba <- plot_bed(yoruba_m1, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
m2_L1PB1_yoruba <- plot_bed(yoruba_m2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
m3_L1PB1_yoruba <- plot_bed(yoruba_m3, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
m1_L1PB1_french <- plot_bed(french_m1, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("French 1")
m2_L1PB1_french <- plot_bed(french_m2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("French 2")
m3_L1PB1_french <- plot_bed(french_m3, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("French 3")
m1_L1PB1_maya <- plot_bed(maya_m1, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
m2_L1PB1_maya <- plot_bed(maya_m2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 2")

m_pop3_L1PB1 <- ggarrange(m1_L1PB1_yoruba, m2_L1PB1_yoruba, m3_L1PB1_yoruba, m1_L1PB1_french, m2_L1PB1_french, m3_L1PB1_french, m1_L1PB1_maya,  m2_L1PB1_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(m_pop3_L1PB1_final <- annotate_figure(m_pop3_L1PB1, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1PB1 - Males", color = "black"), fig.lab = ""))
  

f1_L1PB1_yoruba <- plot_bed(yoruba_f1, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
f2_L1PB1_yoruba <- plot_bed(yoruba_f2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
f3_L1PB1_yoruba <- plot_bed(yoruba_f3, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
f1_L1PB1_french <- plot_bed(french_f1, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("French 1")
f2_L1PB1_french <- plot_bed(french_f2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("French 2")
f3_L1PB1_french <- plot_bed(french_f3, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("French 3")
f1_L1PB1_maya <- plot_bed(maya_f1, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
f2_L1PB1_maya <- plot_bed(maya_f2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 2")
f3_L1PB1_maya <- plot_bed(maya_f2, "L1PB1_te", 'n', 'n', 'n', 'n') + ggtitle("Maya 3")

f_pop3_L1PB1 <- ggarrange(f1_L1PB1_yoruba, f2_L1PB1_yoruba, f3_L1PB1_yoruba, f1_L1PB1_french, f2_L1PB1_french, f3_L1PB1_french, f1_L1PB1_maya,  f2_L1PB1_maya, f3_L1PB1_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(f_pop3_L1PB1_final <- annotate_figure(f_pop3_L1PB1, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1PB1 - Females", color = "black"), fig.lab = ""))
  
```

I conclude that the patterns previously observed are consistent among the different individuals composing the populations.

Now I want to check the scale of the plots to see if the different patterns may be explained like this.
```{r}
m1_L1ME5_yoruba <- plot_bed(yoruba_m1, "L1ME5_te", 'n', 'n', 'y', 'y') + ggtitle("Yoruba male")
m1_L1ME5_french <- plot_bed(french_m1, "L1ME5_te", 'n', 'n', 'y', 'y') + ggtitle("French male") + ylim(0,50)
f1_L1ME5_yoruba <- plot_bed(yoruba_f1, "L1ME5_te", 'n', 'n', 'y', 'y') + ggtitle("Yoruba female")
f1_L1ME5_french <- plot_bed(french_f1, "L1ME5_te", 'n', 'n', 'y', 'y') + ggtitle("French female") + ylim(0,50)

L1ME5_scale <- ggarrange(m1_L1ME5_yoruba, m1_L1ME5_french, f1_L1ME5_yoruba, f1_L1ME5_french, ncol = 2, nrow = 2, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(L1ME5_scale_final <- annotate_figure(L1ME5_scale, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("L1ME5", color = "black"), fig.lab = ""))
```

By setting 50 as limit for the y-axis, the patterns are not so different anymore. I conclude that the main difference is given by a short sequence, mapping around positions 250-300 of the L1ME5 reference sequence and which coverage is enormously different among populations. Now I want to get the exact positions of the interesting sequence.

Thus, I select for the positions in which the **coverage** is `> 1000`. The sequence is around **30 bases long** and it's spanning between positions `274-311`.
```{r}
filter(french_m1, familyname == "L1ME5_te") %>% filter(coverage>1000)

L1ME5_fullsequence <- "ctaaggaaagaattaaaaaaagtacaaaaatatatattcaagaattttcatttcaatattggntttaaaacaataaaatatttggaaacaatctaaatgtctaataatgggagaataaataaattatagtatatctataaaataaaatattatatagctattaaaawtatatttttaaaaaatatttaatgatataaaaaaatntttataatataatattaaataaaaaaaataaattataaaattatatatataatataattttatatnttaaatatataatatatatataaatatatatatatatatatnaaataaaaaaaaaagactgaaaggaaatatacaccaaaatgttaacagtggttatctctgggtggtgggattataggtgatttttatttttttttttttttttatattttctgtattttctaaattttttntaaattttctacaataaatatgtattacttttataatcagaaaaaaaataaaaataataaaaat"

str_sub(L1ME5_fullsequence, start = 274, end = 311) %>% toupper()
```

Aligning the sequence with BLAST did not produce any result. Probably the sequence is too short for this. This question remains: why it is present in `Maya` **females** but not in the **males**? Could it be a **mtDNA** variant which was absent from Africans and then evolved in Eurasians and in some way spread into America? This would explain the pattern in Maya, but not in Eurasians (see `French`) in which this variant is present in both sexes.


### KRAB-ZNF coverage

```{r}
KRAB = "a_ZNF138_5_krab"
m1_a_ZNF138_5_yoruba <- plot_bed(yoruba_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
m2_a_ZNF138_5_yoruba <- plot_bed(yoruba_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
m3_a_ZNF138_5_yoruba <- plot_bed(yoruba_m3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
m1_a_ZNF138_5_french <- plot_bed(french_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 1")
m2_a_ZNF138_5_french <- plot_bed(french_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 2")
m3_a_ZNF138_5_french <- plot_bed(french_m3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 3")
m1_a_ZNF138_5_maya <- plot_bed(maya_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
m2_a_ZNF138_5_maya <- plot_bed(maya_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 2")

m_pop3_a_ZNF138_5 <- ggarrange(m1_a_ZNF138_5_yoruba, m2_a_ZNF138_5_yoruba, m3_a_ZNF138_5_yoruba, m1_a_ZNF138_5_french, m2_a_ZNF138_5_french, m3_a_ZNF138_5_french, m1_a_ZNF138_5_maya,  m2_a_ZNF138_5_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(m_pop3_a_ZNF138_5_final <- annotate_figure(m_pop3_a_ZNF138_5, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("a_ZNF138_5 - Males", color = "black"), fig.lab = ""))
  

f1_a_ZNF138_5_yoruba <- plot_bed(yoruba_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
f2_a_ZNF138_5_yoruba <- plot_bed(yoruba_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
f3_a_ZNF138_5_yoruba <- plot_bed(yoruba_f3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
f1_a_ZNF138_5_french <- plot_bed(french_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 1")
f2_a_ZNF138_5_french <- plot_bed(french_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 2")
f3_a_ZNF138_5_french <- plot_bed(french_f3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 3")
f1_a_ZNF138_5_maya <- plot_bed(maya_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
f2_a_ZNF138_5_maya <- plot_bed(maya_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 2")
f3_a_ZNF138_5_maya <- plot_bed(maya_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 3")

f_pop3_a_ZNF138_5 <- ggarrange(f1_a_ZNF138_5_yoruba, f2_a_ZNF138_5_yoruba, f3_a_ZNF138_5_yoruba, f1_a_ZNF138_5_french, f2_a_ZNF138_5_french, f3_a_ZNF138_5_french, f1_a_ZNF138_5_maya,  f2_a_ZNF138_5_maya, f3_a_ZNF138_5_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(f_pop3_a_ZNF138_5_final <- annotate_figure(f_pop3_a_ZNF138_5, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("a_ZNF138_5 - Females", color = "black"), fig.lab = ""))
  
```


```{r}
KRAB = "a_ZNF718_1_krab"
m1_a_ZNF718_1_yoruba <- plot_bed(yoruba_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
m2_a_ZNF718_1_yoruba <- plot_bed(yoruba_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
m3_a_ZNF718_1_yoruba <- plot_bed(yoruba_m3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
m1_a_ZNF718_1_french <- plot_bed(french_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 1")
m2_a_ZNF718_1_french <- plot_bed(french_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 2")
m3_a_ZNF718_1_french <- plot_bed(french_m3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 3")
m1_a_ZNF718_1_maya <- plot_bed(maya_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
m2_a_ZNF718_1_maya <- plot_bed(maya_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 2")

m_pop3_a_ZNF718_1 <- ggarrange(m1_a_ZNF718_1_yoruba, m2_a_ZNF718_1_yoruba, m3_a_ZNF718_1_yoruba, m1_a_ZNF718_1_french, m2_a_ZNF718_1_french, m3_a_ZNF718_1_french, m1_a_ZNF718_1_maya,  m2_a_ZNF718_1_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(m_pop3_a_ZNF718_1_final <- annotate_figure(m_pop3_a_ZNF718_1, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("a_ZNF718_1 - Males", color = "black"), fig.lab = ""))
  

f1_a_ZNF718_1_yoruba <- plot_bed(yoruba_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
f2_a_ZNF718_1_yoruba <- plot_bed(yoruba_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
f3_a_ZNF718_1_yoruba <- plot_bed(yoruba_f3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
f1_a_ZNF718_1_french <- plot_bed(french_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 1")
f2_a_ZNF718_1_french <- plot_bed(french_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 2")
f3_a_ZNF718_1_french <- plot_bed(french_f3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 3")
f1_a_ZNF718_1_maya <- plot_bed(maya_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
f2_a_ZNF718_1_maya <- plot_bed(maya_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 2")
f3_a_ZNF718_1_maya <- plot_bed(maya_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 3")

f_pop3_a_ZNF718_1 <- ggarrange(f1_a_ZNF718_1_yoruba, f2_a_ZNF718_1_yoruba, f3_a_ZNF718_1_yoruba, f1_a_ZNF718_1_french, f2_a_ZNF718_1_french, f3_a_ZNF718_1_french, f1_a_ZNF718_1_maya,  f2_a_ZNF718_1_maya, f3_a_ZNF718_1_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(f_pop3_a_ZNF718_1_final <- annotate_figure(f_pop3_a_ZNF718_1, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("a_ZNF718_1 - Females", color = "black"), fig.lab = ""))
  
```

```{r}
KRAB = "a_ZNF844_1_krab"
m1_a_ZNF844_1_yoruba <- plot_bed(yoruba_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
m2_a_ZNF844_1_yoruba <- plot_bed(yoruba_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
m3_a_ZNF844_1_yoruba <- plot_bed(yoruba_m3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
m1_a_ZNF844_1_french <- plot_bed(french_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 1")
m2_a_ZNF844_1_french <- plot_bed(french_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 2")
m3_a_ZNF844_1_french <- plot_bed(french_m3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 3")
m1_a_ZNF844_1_maya <- plot_bed(maya_m1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
m2_a_ZNF844_1_maya <- plot_bed(maya_m2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 2")

m_pop3_a_ZNF844_1 <- ggarrange(m1_a_ZNF844_1_yoruba, m2_a_ZNF844_1_yoruba, m3_a_ZNF844_1_yoruba, m1_a_ZNF844_1_french, m2_a_ZNF844_1_french, m3_a_ZNF844_1_french, m1_a_ZNF844_1_maya,  m2_a_ZNF844_1_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(m_pop3_a_ZNF844_1_final <- annotate_figure(m_pop3_a_ZNF844_1, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("a_ZNF844_1 - Males", color = "black"), fig.lab = ""))
  

f1_a_ZNF844_1_yoruba <- plot_bed(yoruba_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 1")
f2_a_ZNF844_1_yoruba <- plot_bed(yoruba_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 2")
f3_a_ZNF844_1_yoruba <- plot_bed(yoruba_f3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Yoruba 3")
f1_a_ZNF844_1_french <- plot_bed(french_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 1")
f2_a_ZNF844_1_french <- plot_bed(french_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 2")
f3_a_ZNF844_1_french <- plot_bed(french_f3, KRAB, 'n', 'n', 'n', 'n') + ggtitle("French 3")
f1_a_ZNF844_1_maya <- plot_bed(maya_f1, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 1")
f2_a_ZNF844_1_maya <- plot_bed(maya_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 2")
f3_a_ZNF844_1_maya <- plot_bed(maya_f2, KRAB, 'n', 'n', 'n', 'n') + ggtitle("Maya 3")

f_pop3_a_ZNF844_1 <- ggarrange(f1_a_ZNF844_1_yoruba, f2_a_ZNF844_1_yoruba, f3_a_ZNF844_1_yoruba, f1_a_ZNF844_1_french, f2_a_ZNF844_1_french, f3_a_ZNF844_1_french, f1_a_ZNF844_1_maya,  f2_a_ZNF844_1_maya, f3_a_ZNF844_1_maya, ncol = 3, nrow = 3, common.legend = TRUE, legend = "bottom", align = "hv", font.label = list(size = 10, color = "black", face = "bold", family = NULL, position = "top"))

(f_pop3_a_ZNF844_1_final <- annotate_figure(f_pop3_a_ZNF844_1, left = text_grob("Coverage", color = "black", rot = 90), bottom = text_grob("Base number", color = "black"), top = text_grob("a_ZNF844_1 - Females", color = "black"), fig.lab = ""))
  
```

## Are there unmapped reads in the .cram file?

Another important thing to check is if **unmapped reads** are present in the initial **.cram file**. If this is not the case, many reads with potentially interesting information would just be lost just because they do not align on the reference genome.

To check for this, we can use this UNIX command giving the .cram file path.
```
samtools view HGDP00001-Brahui.cram | cut -f 3
```

In our dataset, the unmapped reads (identified with a `*` instead of the number of chromosome) are present.


## Check scgx coverage

```{r}
scgx <- filter(HGDPcutoff, type=="scgx") %>% group_by(ID, sex, Country) %>% dplyr::summarise(mean_scgx = mean(copynumber))

ggplot(scgx, aes(ID, mean_scgx, color=sex))+
  geom_point()

ggplot(scgx, aes(ID, mean_scgx, color=Country))+
  geom_point() + facet_grid(~ sex)
```

Few outliers are present in both sexes, but in general the expected coverage (0.5 for males, 1 for females) is matched.