---
title: "Why the results in the two datasets are that different?"
output: rmarkdown::github_document
editor_options: 
  markdown: 
    wrap: sentence
---

```{r}
library(tidyverse)
```


## Check the amount of unmapped reads in both the datasets

We expect the amount of unmapped reads in the two datasets to be slightly different due to different sequencing methods and read length. If the differences are huge, it's not expected. We downloaded 3 samples for each dataset.

This command subset the full cram file into a file containing only the unmapped reads.
```
samtools view -f 4 /Volumes/Temp2/rpianezza/testing_unmapped_reads/HGDP01303.alt_bwamem_GRCh38DH.20181023.Uygur.cram >> /Volumes/Temp2/rpianezza/testing_unmapped_reads/unmapped_HGDP_HGDP01303
```

This create an output file with the number of reads of each cram file.
```
find /Volumes/Temp2/rpianezza/testing_unmapped_reads -name '*.cram' -exec sh -c 'echo $(samtools view -c {}) $(basename {}) >> /Volumes/Temp2/rpianezza/testing_unmapped_reads/wc.tsv' \; 
```


## Run the pipeline on FASTQ files

Our program is based on processing .cram files, the aligned file to the human reference genome. The unmapped reads were present in the files, so the fact that the file is aligned should not have an impact on our results. Anyway, here we run the pipeline on  the .fastq files for 3 african females and 3 european females and compare the results with the .cram files. We do this for both the datasets.

First we download all the .fastq files for each sample and merge them in a single fastq file. Then, using a modified version of the mapping script (run.map.fastqp), we run the pipeline on these files.
```
cat /Volumes/Temp2/riccardo/fastq-control/ERR757740_1.fastq.gz /Volumes/Temp2/riccardo/fastq-control/ERR757740_2.fastq.gz >> /Volumes/Temp2/riccardo/fastq-control/full.fastq.files/HGDP00471_ERR757740.fastq.gz 
```


## How many unmapped reads do we expect?

Mapping the artificial reads with different read lengths (100 and 150 for SGDP and HGDP) to the Human reference genome 38, the one used for the .cram files production.

Create artificial reads: 
* female, 100 rl, 30x coverage
* male, 100, rl, 30x coverage (first splitted in x-autosomes and y-autosomes, then merged)

The same files for read length 150 were already present.
```
python /Volumes/Temp1/rpianezza/TE/human-te-dynamics-svn/scripts/create-reads-for-human.py --fasta /Volumes/Temp1/rpianezza/T2T/sexes/autosomes_x --coverage 30 --read-length 100 --output /Volumes/Temp1/rpianezza/T2T/T2T-artificial-reads/female.100rl.fastq.gz --method uniform
```


## Sequencing libraries bias

PCR and PCR-free libraries.

```{r}
HGDP <- read_delim("/Volumes/Temp1/rpianezza/TE/summary-HGDP/USEME_HGDP_mq0_cutoff0.01.txt")
names(HGDP)<-c("ID","pop","sex","country","type","familyname","length","reads","copynumber","batch")

(libraries <- read_tsv("/Volumes/Temp1/rpianezza/investigation/libraries_HGDP_by_pop.tsv", col_names = c("pop", "country", "latitude", "longitude", "PCR_free", "PCR"), skip=1) %>% select(c(pop, latitude, longitude, PCR_free, PCR)))

by_pop <- filter(HGDP, type=='te') %>% group_by(pop, country, familyname, sex) %>% dplyr::summarise(sd=sd(copynumber), copynumber = mean(copynumber), count=n())

data <- inner_join(libraries, by_pop, by = "pop") %>% type_convert()
```

```{r}
plot_map <- function(data, famname){
TE <- filter(data, familyname == famname)
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(
    data = TE, aes(x=longitude, y=latitude, color = copynumber, size = count)
  ) + geom_errorbar() + theme(legend.position="top") + scale_colour_gradient(low = "green", high = "red") + theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(~sex) + ggtitle(famname)}

plot_libraries <- function(data, famname){
TE <- data %>% select(c(pop, latitude, longitude, PCR_free, PCR, sex, count)) %>% mutate(pcr_freq = PCR/(PCR_free+PCR))
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(
    data = TE, aes(x=longitude, y=latitude, color = pcr_freq, size = count)
  ) + geom_errorbar() + theme(legend.position="top") + scale_colour_gradient(low = "green", high = "red") + theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(~sex) + ggtitle(famname)}
```

```{r}
plot_map(data, "L1PA16")
plot_libraries(data, "PCR-libraries")
```

```{r}
(count <- libraries %>% type_convert() %>% dplyr::summarise(PCR_free = sum(PCR_free), PCR = sum(PCR)))
libraries
#no_pcr <- read_tsv("/Volumes/Temp1/rpianezza/investigation/HGDP-only-pcr-free-samples.tsv", col_names = c("url",	"md5",	"Data_collection", "Data_type", "Analysis_group", "Sample", "Population", "Data reuse policy")) %>% select(Sample)

#write_tsv(no_pcr, "/Volumes/Temp1/rpianezza/investigation/HGDP-only-pcr-free-samples.tsvx")
```