---
title: "Why the results in the two datasets are that different?"
output: rmarkdown::github_document
editor_options: 
  markdown: 
    wrap: sentence
---

After processing the SGDP dataset, the results were ambiguous. While the most variable TEs are confirmed in both the datasets as well as the most divergent among sexes, the geographical pattern disappeared both in the world maps and in the copynumbers PCA. Let's try to understand why.
```{r}
library(tidyverse)
```

## Check the amount of unmapped reads in both the datasets

We expect the amount of unmapped reads in the two datasets to be slightly different due to different sequencing methods and read length. If the differences are huge, it's not expected. We downloaded 3 samples for each dataset.

This command subset the full cram file into a file containing only the unmapped reads.
```
samtools view -f 4 /Volumes/Temp2/rpianezza/testing_unmapped_reads/HGDP01303.alt_bwamem_GRCh38DH.20181023.Uygur.cram >> /Volumes/Temp2/rpianezza/testing_unmapped_reads/unmapped_HGDP_HGDP01303
```

This create an output file with the number of reads of each cram file.
```
find /Volumes/Temp2/rpianezza/testing_unmapped_reads -name '*.cram' -exec sh -c 'echo $(samtools view -c {}) $(basename {}) >> /Volumes/Temp2/rpianezza/testing_unmapped_reads/wc.tsv' \; 
```

I manually conclude the file that I read here:
```{r}
(unmapped <- read_tsv("/Volumes/Temp2/rpianezza/testing_unmapped_reads/wc.tsv") %>% mutate(unmapped = all-mapped) %>% mutate(freq_unmapped = unmapped/all))
```

The differences are small, and probably explainable from the different rl.


## Sequencing libraries bias

In the HGDP supplement, it's reported that some samples are produced using PCR, while the vast majority is PCR-free. Let's see if it has an impact.

```{r}
HGDP <- read_delim("/Volumes/Temp1/rpianezza/TE/summary-HGDP/USEME_HGDP_mq0_cutoff0.01.txt")
names(HGDP)<-c("ID","pop","sex","country","type","familyname","length","reads","copynumber","batch")

(libraries <- read_tsv("/Volumes/Temp1/rpianezza/investigation/libraries_HGDP_by_pop.tsv", col_names = c("pop", "country", "latitude", "longitude", "PCR_free", "PCR"), skip=1) %>% select(c(pop, latitude, longitude, PCR_free, PCR)))

by_pop <- filter(HGDP, type=='te') %>% group_by(pop, country, familyname, sex) %>% dplyr::summarise(sd=sd(copynumber), copynumber = mean(copynumber), count=n())

data <- inner_join(libraries, by_pop, by = "pop") %>% type_convert()
```

```{r}
plot_map <- function(data, famname){
TE <- filter(data, familyname == famname)
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(
    data = TE, aes(x=longitude, y=latitude, color = copynumber, size = count)
  ) + geom_errorbar() + theme(legend.position="top") + scale_colour_gradient(low = "green", high = "red") + theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(~sex) + ggtitle(famname)}

plot_libraries <- function(data, famname){
TE <- data %>% select(c(pop, latitude, longitude, PCR_free, PCR, sex, count)) %>% mutate(pcr_freq = PCR/(PCR_free+PCR))
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(
    data = TE, aes(x=longitude, y=latitude, color = pcr_freq, size = count)
  ) + geom_errorbar() + theme(legend.position="top") + scale_colour_gradient(low = "green", high = "red") + theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(~sex) + ggtitle(famname)}
```

```{r}
plot_map(data, "L1PA16")
plot_libraries(data, "PCR-libraries")
```

Wow. We can conclude that at, least partially, our geographic pattern IS influenced by this. I can filter the PCR samples from both the datasets (around 100 in HGDP, 16 in SGDP).

```{r}
no_pcr <- read_tsv("/Volumes/Temp1/rpianezza/investigation/SGDP_no_PCR.tsv", col_names = c("url",	"md5",	"Data_collection", "Data_type", "Analysis_group", "Sample", "Population", "Data reuse policy")) %>% select(Sample)

#write_tsv(no_pcr, "/Volumes/Temp1/rpianezza/investigation/SGDP-only-pcr-free-samples.tsv")
```

```{r}
sgdp_pcr_free <- read_tsv("/Volumes/Temp1/rpianezza/investigation/SGDP-only-pcr-free-samples.tsv", col_names = "sample")

sgdp_metadata <- read_tsv("/Volumes/Temp1/rpianezza/SGDP/metadata/SGDP_metadata.tsv") %>% select(sample, biosample)

sgdp_pcr_free_samples <- inner_join(sgdp_pcr_free, sgdp_metadata, by="sample") %>% select(biosample) %>% dplyr::rename(ID=biosample)

#write_tsv(sgdp_pcr_free_samples, "/Volumes/Temp1/rpianezza/investigation/SGDP-no-PCR/SGDP-no-PCR.tsv")
```

## Check the quality of sequencing in the two dataset

By calculating variance of scg copynumbers in each sample, we can estimate the precision of the sequencing method used.

```{r}
SGDP <- read_tsv("/Volumes/Temp2/rpianezza/SGDP/summary/USEME_SGDP_cutoff_mq0")
HGDP <- read_delim("/Volumes/Temp1/rpianezza/TE/summary-HGDP/USEME_HGDP_mq0_cutoff0.01.txt")
names(HGDP) <- c("ID","pop","sex","country","type","familyname","length","reads","copynumber","batch")
HGDP <- HGDP %>% mutate(country = recode(country, "Oceania_(SGDP),Oceania"="Oceania"))
```

```{r}
(SGDP_scg <- SGDP %>% filter(type=="scg") %>% group_by(biosample, pop, country, sex) %>% dplyr::summarise(var = var(copynumber)) %>% arrange(desc(var)))# %>% filter(var<0.025))

(HGDP_scg <- HGDP %>% filter(type=="scg") %>% group_by(ID, pop, country, sex) %>% dplyr::summarise(var = var(copynumber)) %>% arrange(desc(var)))# %>% filter(var<0.025))
```

We conclude that the HGDP samples have on average higher quality, since only 2 samples have a variability in SCGs copynumber > 0.2, compared to the SGDP where there are 26 samples over this threshold!


## Run the pipeline on FASTQ files

Our program is based on processing .cram files, the aligned file to the human reference genome. The unmapped reads were present in the files, so the fact that the file is aligned should not have an impact on our results. Anyway, here we run the pipeline on  the .fastq files for 3 african females and 3 european females and compare the results with the .cram files. We do this for both the datasets.

First we download all the .fastq files for each sample and merge them in a single fastq file. Then, using a modified version of the mapping script (run.map.fastqp), we run the pipeline on these files.
```
cat /Volumes/Temp2/riccardo/fastq-control/ERR757740_1.fastq.gz /Volumes/Temp2/riccardo/fastq-control/ERR757740_2.fastq.gz >> /Volumes/Temp2/riccardo/fastq-control/full.fastq.files/HGDP00471_ERR757740.fastq.gz 
```



