---
title: "Archaic humans - SNPs found in NH and DH"
output: rmarkdown::github_document
editor_options: 
  markdown: 
    wrap: sentence
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '/Users/rpianezza/TE/')
```

Set the environment and creating metadata file.
```{r}
library(tidyverse)
library(umap)

HGDP <- read_delim("/Volumes/Temp1/rpianezza/0.old/summary-HGDP/HGDP_cutoff_classified.tsv") %>% select(ID, sex, country, pop) %>% distinct()

archaic <- tibble(
  ID = c("Vindija", "Altai", "Denisovan"),
  pop = c("Vindija", "Altai", "Denisova"),
  country = c("Neandertal", "Neandertal", "Denisovan"),
  sex = c("female", "female", "female")
)

HGDP_arch <- bind_rows(HGDP, archaic)

coordinates <- read_tsv("/Users/rpianezza/TE/summary-HGDP/HGDP_populationcoordinates.txt", col_names = c("pop", "region", "latitude", "longitude")) %>% select(pop, latitude, longitude)
```

```{r}
afr <- HGDP %>% filter(country=="Africa", sex=="female") %>% select(ID)
#write_tsv(afr, "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/ID-Afr-females")

fem <- HGDP %>% filter(sex=="female") %>% select(ID)
#write_tsv(fem, "/Volumes/Temp1/rpianezza/PCA-SNPs-all-analysis/matrixes/females/females.IDs")

mal <- HGDP %>% filter(sex=="male") %>% select(ID)
#write_tsv(mal, "/Volumes/Temp1/rpianezza/PCA-SNPs-all-analysis/matrixes/females/males.IDs")
```


## SCGs

Reading the matrixes for the archaic variants in SCGs in the HGDP. 1 file for each archaic genome.
```{r scg-matrixes}
scg_vindija_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/vindija1k")
scg_altai_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/altai1k")
scg_denisova_all <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/denisova1k")
```

Read the files containing info about the variants called in the archaic genome. Is important the "total_diff" column, which is an indicator of the distance between the allele frequencies in the Africans and in the archaic genome.

Is comprised between 0 (same allele frequencies for that position) and 2 (complete non-overlapping allele frequencies for that position).

```{r scg-variants}
scg_diff_vindija <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/vindija1k_afr-arch") %>% select(familyname, position, total_diff)
scg_diff_altai <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/altai1k_afr-arch") %>% select(familyname, position, total_diff)
scg_diff_denisova <- read_tsv("/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/archaic-variants/scg/1k/denisova1k_afr-arch") %>% select(familyname, position, total_diff)
```

Merge the two files read above for each of the three genome and select only the variance with a **total_diff** > a certain threshold. In this case, I use 1.75. This means that, for single copy genes, both the alleles must be different between archaic and africans to keep the variant in this code. Ideally, this will result in a total_diff = 2, but due to sequencing error and aDNA bias I use a lower threshold (1.97).

```{r scg-selected-variants}
scg_vindija <- inner_join(scg_vindija_all, scg_diff_vindija, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
scg_altai <- inner_join(scg_altai_all, scg_diff_altai, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
scg_denisova <- inner_join(scg_denisova_all, scg_diff_denisova, by=c("familyname", "position")) %>% filter(total_diff>1.97) %>% relocate(total_diff) %>% arrange(desc(total_diff))
```

```{r}
compute_blocks <- function(x) {
  num_entries <- sum(x < 1.25)
  # Identify blocks of values less than 1.25
  blocks <- rle(x < 1.25)
  # Count the number of blocks
  num_blocks <- sum(blocks$values)
  # Compute the mean length of blocks
  mean_length <- mean(blocks$lengths[blocks$values])
  # Return a named list with the results
  list(num_entries = num_entries, num_blocks = num_blocks, mean_length = mean_length)
}

blocks <- function(data) {
data_num <- select(data, -c(familyname, position, total_diff))
result <- summarise_all(data_num, compute_blocks) %>% unnest(everything()) %>% t() %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% dplyr::rename("introgressed" = "V1", "blocks" = "V2", "mean_block_len" = "V3") %>% as_tibble()
}
```

```{r}
(vindija_hap <- blocks(scg_vindija))
(altai_hap <- blocks(scg_altai))
(denisova_hap <- blocks(scg_denisova))
```



