Archaic humans - SNPs found in NH and DH
================

Set the environment and creating metadata file.

``` r
library(tidyverse)
```

    ## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
    ## ✔ dplyr     1.1.0     ✔ readr     2.1.4
    ## ✔ forcats   1.0.0     ✔ stringr   1.5.0
    ## ✔ ggplot2   3.4.1     ✔ tibble    3.2.0
    ## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
    ## ✔ purrr     1.0.1     
    ## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()
    ## ℹ Use the ]8;;http://conflicted.r-lib.org/conflicted package]8;; to force all conflicts to become errors

``` r
library(umap)

HGDP <- read_delim("/Volumes/Temp1/rpianezza/0.old/summary-HGDP/HGDP_cutoff_classified.tsv") %>% select(ID, sex, country, pop) %>% distinct()
```

    ## Rows: 1394352 Columns: 12
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (9): ID, pop, sex, country, type, familyname, batch, superfamily, shared...
    ## dbl (3): length, reads, copynumber
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
archaic <- tibble(
  ID = c("Vindija", "Altai", "Denisovan"),
  pop = c("Vindija", "Altai", "Denisova"),
  country = c("Neandertal", "Neandertal", "Denisovan"),
  sex = c("female", "female", "female")
)

HGDP_arch <- bind_rows(HGDP, archaic)

HGDP_fem <- HGDP_arch %>% filter(sex=="female")

coordinates <- read_tsv("/Users/rpianezza/TE/summary-HGDP/HGDP_populationcoordinates.txt", col_names = c("pop", "region", "latitude", "longitude")) %>% select(pop, latitude, longitude)
```

    ## Rows: 54 Columns: 4
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr (2): pop, region
    ## dbl (2): latitude, longitude
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

## TEs

Base order: A T C G

``` r
vindija <- "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/diagnostic-variants/vindija"
vindija_diag <- "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/diagnostic-variants/vindija_afr-arch"

altai <- "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/diagnostic-variants/altai"
altai_diag <- "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/diagnostic-variants/altai_afr-arch"

denisova <- "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/diagnostic-variants/denisova"
denisova_diag <- "/Volumes/Temp1/rpianezza/ancient_humans/archaic-humans/analysis/diagnostic-variants/denisova_afr-arch"
```

``` r
plot_introgression <- function(data, titlee){
  
 plot <- data %>% group_by(pop, country) %>% dplyr::summarise(mean_intro = mean(introgressed), sd_intro = sd(introgressed)) %>% ggplot(aes(x = reorder(pop, -mean_intro), y=mean_intro, fill=country))+
  geom_col()+
  geom_errorbar(aes(ymin = mean_intro - sd_intro, ymax = mean_intro + sd_intro), 
                width = 0.2, position = position_dodge(0.9), alpha = 0.5)+
  labs(x = "", y = "Introgressed variants", title = element_text(face = "bold", size = 14, titlee)) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "bottom", plot.title = element_text(hjust = 0.5))
}


plot_map <- function(intro, coord, title){
  
full <- inner_join(intro, coord, by="pop") %>% group_by(pop, longitude, latitude) %>% dplyr::summarise(introgressed = mean(introgressed), count = n())
world_map = map_data("world")

ggplot() +
  geom_map(
    data = world_map, map = world_map,
    aes(long, lat, map_id = region),
    color = "white", fill = "lightgray", size = 0) +
  geom_point(data = full, aes(x=longitude, y=latitude, color = introgressed, size = count)) + geom_errorbar() + scale_colour_gradient(low = "green", high = "red") + theme(legend.position="top") + theme(plot.title = element_text(hjust = 0.5)) + ggtitle(title)
}


introgression_diagnostic <- function(matrix, diagnostic_bases, coord, titlee){
  
archaic <- read_tsv(matrix) %>% select(-diagnostic, -afr_copynumber)
diag <- read_tsv(diagnostic_bases) %>% select(familyname, position, afr_copynumber, diagnostic)

merged <- inner_join(archaic, diag, by=c("familyname", "position")) %>% relocate(diagnostic, afr_copynumber) %>% mutate_at(vars(starts_with("HGDP")), ~. * afr_copynumber) %>% filter(afr_copynumber<250) %>% select(-c(diagnostic, afr_copynumber, familyname, position)) %>% dplyr::summarise_all(~sum(. >1, na.rm=TRUE)) %>% t %>% as.data.frame() %>% rownames_to_column(var = "ID") %>% as_tibble() %>% dplyr::rename("introgressed" = "V1") %>% separate(ID, into=c("ID", "pop"), sep="-", remove=FALSE) %>% select(-pop) %>% inner_join(HGDP_fem, by="ID")

bar <- plot_introgression(merged, titlee)
mappa <- plot_map(merged, coord, titlee)

list(barplot = bar, map = mappa)
}
```

``` r
introgression_diagnostic(vindija, vindija_diag, coordinates, "Variants introgressed (Vindija) - RepSeq")
```

    ## Warning: One or more parsing issues, call `problems()` on your data frame for details,
    ## e.g.:
    ##   dat <- vroom(...)
    ##   problems(dat)

    ## Rows: 3326 Columns: 832
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## lgl   (2): afr_copynumber, diagnostic
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
    ## Rows: 3266 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (2): familyname, diagnostic
    ## dbl (11): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

    ## Warning in inner_join(archaic, diag, by = c("familyname", "position")): Each row in `x` is expected to match at most 1 row in `y`.
    ## ℹ Row 116 of `x` matches multiple rows.
    ## ℹ If multiple matches are expected, set `multiple = "all"` to silence this
    ##   warning.

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.
    ## `summarise()` has grouped output by 'pop', 'longitude'. You can override using
    ## the `.groups` argument.

    ## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
    ## ℹ Please use `linewidth` instead.

    ## Warning in geom_map(data = world_map, map = world_map, aes(long, lat, map_id =
    ## region), : Ignoring unknown aesthetics: x and y

    ## $barplot

![](04.archaic-SNPs-diagnostic_files/figure-gfm/unnamed-chunk-2-1.png)<!-- -->

    ## 
    ## $map

![](04.archaic-SNPs-diagnostic_files/figure-gfm/unnamed-chunk-2-2.png)<!-- -->

``` r
introgression_diagnostic(altai, altai_diag, coordinates, "Variants introgressed (Altai) - RepSeq")
```

    ## Warning: One or more parsing issues, call `problems()` on your data frame for details,
    ## e.g.:
    ##   dat <- vroom(...)
    ##   problems(dat)

    ## Rows: 1059 Columns: 832
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## lgl   (2): afr_copynumber, diagnostic
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
    ## Rows: 999 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (2): familyname, diagnostic
    ## dbl (11): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

    ## Warning in inner_join(archaic, diag, by = c("familyname", "position")): Each row in `x` is expected to match at most 1 row in `y`.
    ## ℹ Row 38 of `x` matches multiple rows.
    ## ℹ If multiple matches are expected, set `multiple = "all"` to silence this
    ##   warning.

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.
    ## `summarise()` has grouped output by 'pop', 'longitude'. You can override using
    ## the `.groups` argument.

    ## Warning in geom_map(data = world_map, map = world_map, aes(long, lat, map_id =
    ## region), : Ignoring unknown aesthetics: x and y

    ## $barplot

![](04.archaic-SNPs-diagnostic_files/figure-gfm/unnamed-chunk-2-3.png)<!-- -->

    ## 
    ## $map

![](04.archaic-SNPs-diagnostic_files/figure-gfm/unnamed-chunk-2-4.png)<!-- -->

``` r
introgression_diagnostic(denisova, denisova_diag, coordinates, "Variants introgressed (Denisovans) - RepSeq")
```

    ## Warning: One or more parsing issues, call `problems()` on your data frame for details,
    ## e.g.:
    ##   dat <- vroom(...)
    ##   problems(dat)

    ## Rows: 958 Columns: 832
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## lgl   (2): afr_copynumber, diagnostic
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
    ## Rows: 912 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (2): familyname, diagnostic
    ## dbl (11): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

    ## Warning in inner_join(archaic, diag, by = c("familyname", "position")): Each row in `x` is expected to match at most 1 row in `y`.
    ## ℹ Row 42 of `x` matches multiple rows.
    ## ℹ If multiple matches are expected, set `multiple = "all"` to silence this
    ##   warning.

    ## `summarise()` has grouped output by 'pop'. You can override using the `.groups`
    ## argument.
    ## `summarise()` has grouped output by 'pop', 'longitude'. You can override using
    ## the `.groups` argument.

    ## Warning in geom_map(data = world_map, map = world_map, aes(long, lat, map_id =
    ## region), : Ignoring unknown aesthetics: x and y

    ## $barplot

![](04.archaic-SNPs-diagnostic_files/figure-gfm/unnamed-chunk-2-5.png)<!-- -->

    ## 
    ## $map

![](04.archaic-SNPs-diagnostic_files/figure-gfm/unnamed-chunk-2-6.png)<!-- -->

``` r
diagnostic_SNPs <- function(matrix, diagnostic_bases){
archaic <- read_tsv(matrix) %>% select(-diagnostic, -afr_copynumber)
diag <- read_tsv(diagnostic_bases) %>% select(familyname, position, afr_copynumber, diagnostic)

(snps <- inner_join(archaic, diag, by=c("familyname", "position")) %>% relocate(diagnostic, afr_copynumber) %>% mutate_at(vars(starts_with("HGDP")), ~. * afr_copynumber) %>% filter(afr_copynumber<250) %>% distinct())
}


diagnostic_SNPs(vindija, vindija_diag)
```

    ## Warning: One or more parsing issues, call `problems()` on your data frame for details,
    ## e.g.:
    ##   dat <- vroom(...)
    ##   problems(dat)

    ## Rows: 3326 Columns: 832
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## lgl   (2): afr_copynumber, diagnostic
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
    ## Rows: 3266 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (2): familyname, diagnostic
    ## dbl (11): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

    ## Warning in inner_join(archaic, diag, by = c("familyname", "position")): Each row in `x` is expected to match at most 1 row in `y`.
    ## ℹ Row 116 of `x` matches multiple rows.
    ## ℹ If multiple matches are expected, set `multiple = "all"` to silence this
    ##   warning.

    ## # A tibble: 3,167 × 832
    ##    diagnostic afr_copy…¹ famil…² posit…³ HGDP0…⁴ HGDP0…⁵ HGDP0…⁶ HGDP0…⁷ HGDP0…⁸
    ##    <chr>           <dbl> <chr>     <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>
    ##  1 G                20.9 HERVK3…     249       0       0   0.837   0.673   0.248
    ##  2 T                20.9 HERVK3…    1626       0       0   0       0       0    
    ##  3 T                20.9 HERVK3…    1636       0       0   0       0       0    
    ##  4 T                20.9 HERVK3…    2239       0       0   0       0       0    
    ##  5 A                20.9 HERVK3…    2898       0       0   0       0       0    
    ##  6 A                20.9 HERVK3…    4327       0       0   0       0       0    
    ##  7 T                20.9 HERVK3…    4331       0       0   0       0       0    
    ##  8 A                20.9 HERVK3…    4662       0       0   0       0       0    
    ##  9 T                20.9 HERVK3…    5359       0       0   0       0       0    
    ## 10 T                20.9 HERVK3…    5371       0       0   0       0       0    
    ## # … with 3,157 more rows, 823 more variables: `HGDP00011-Brahui` <dbl>,
    ## #   `HGDP00013-Brahui` <dbl>, `HGDP00015-Brahui` <dbl>,
    ## #   `HGDP00017-Brahui` <dbl>, `HGDP00021-Brahui` <dbl>,
    ## #   `HGDP00023-Brahui` <dbl>, `HGDP00025-Brahui` <dbl>,
    ## #   `HGDP00029-Brahui` <dbl>, `HGDP00031-Brahui` <dbl>,
    ## #   `HGDP00033-Brahui` <dbl>, `HGDP00035-Brahui` <dbl>,
    ## #   `HGDP00037-Brahui` <dbl>, `HGDP00039-Brahui` <dbl>, …

``` r
diagnostic_SNPs(altai, altai_diag)
```

    ## Warning: One or more parsing issues, call `problems()` on your data frame for details,
    ## e.g.:
    ##   dat <- vroom(...)
    ##   problems(dat)

    ## Rows: 1059 Columns: 832
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## lgl   (2): afr_copynumber, diagnostic
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
    ## Rows: 999 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (2): familyname, diagnostic
    ## dbl (11): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

    ## Warning in inner_join(archaic, diag, by = c("familyname", "position")): Each row in `x` is expected to match at most 1 row in `y`.
    ## ℹ Row 38 of `x` matches multiple rows.
    ## ℹ If multiple matches are expected, set `multiple = "all"` to silence this
    ##   warning.

    ## # A tibble: 823 × 832
    ##    diagnostic afr_copy…¹ famil…² posit…³ HGDP0…⁴ HGDP0…⁵ HGDP0…⁶ HGDP0…⁷ HGDP0…⁸
    ##    <chr>           <dbl> <chr>     <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>
    ##  1 C                7.02 LTR65_…      79    0     0        0       0      0     
    ##  2 G               20.9  HERVK3…     249    0     0        0.837   0.673  0.248 
    ##  3 G               20.9  HERVK3…    5706    0     1.12     0       0      0     
    ##  4 G               20.9  HERVK3…    6199    0     0        0       0      0     
    ##  5 A               20.9  HERVK3…    6716    0     0.0223   0       0      0.0276
    ##  6 G              134.   HERV9_…    6744    1.18  0.928    1.49    1.16   0.681 
    ##  7 T               86.6  L1PA12…     910    0     0        0       0      0     
    ##  8 C               86.6  L1PA12…    1036    0     0        0       0      0     
    ##  9 A               86.6  L1PA12…    1037    0     0        0       0      0     
    ## 10 G               86.6  L1PA12…    1039    0     0        0       0      0     
    ## # … with 813 more rows, 823 more variables: `HGDP00011-Brahui` <dbl>,
    ## #   `HGDP00013-Brahui` <dbl>, `HGDP00015-Brahui` <dbl>,
    ## #   `HGDP00017-Brahui` <dbl>, `HGDP00021-Brahui` <dbl>,
    ## #   `HGDP00023-Brahui` <dbl>, `HGDP00025-Brahui` <dbl>,
    ## #   `HGDP00029-Brahui` <dbl>, `HGDP00031-Brahui` <dbl>,
    ## #   `HGDP00033-Brahui` <dbl>, `HGDP00035-Brahui` <dbl>,
    ## #   `HGDP00037-Brahui` <dbl>, `HGDP00039-Brahui` <dbl>, …

``` r
diagnostic_SNPs(denisova, denisova_diag)
```

    ## Warning: One or more parsing issues, call `problems()` on your data frame for details,
    ## e.g.:
    ##   dat <- vroom(...)
    ##   problems(dat)

    ## Rows: 958 Columns: 832
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr   (1): familyname
    ## dbl (829): position, HGDP00001-Brahui, HGDP00003-Brahui, HGDP00005-Brahui, H...
    ## lgl   (2): afr_copynumber, diagnostic
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
    ## Rows: 912 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: "\t"
    ## chr  (2): familyname, diagnostic
    ## dbl (11): position, A_x, T_x, C_x, G_x, A_y, T_y, C_y, G_y, afr_copynumber, ...
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

    ## Warning in inner_join(archaic, diag, by = c("familyname", "position")): Each row in `x` is expected to match at most 1 row in `y`.
    ## ℹ Row 42 of `x` matches multiple rows.
    ## ℹ If multiple matches are expected, set `multiple = "all"` to silence this
    ##   warning.

    ## # A tibble: 752 × 832
    ##    diagnostic afr_copy…¹ famil…² posit…³ HGDP0…⁴ HGDP0…⁵ HGDP0…⁶ HGDP0…⁷ HGDP0…⁸
    ##    <chr>           <dbl> <chr>     <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>
    ##  1 G                20.9 HERVK3…     149   0       0       0       0       0    
    ##  2 G                20.9 HERVK3…     249   0       0       0.837   0.673   0.248
    ##  3 A               134.  HERV9_…    5870   0       0       0       0       0    
    ##  4 A               134.  HERV9_…    6860   0       0       0       0       0    
    ##  5 A                86.6 L1PA12…    1042   0       0.367   0       0       0    
    ##  6 A                16.5 LTR27C…     619   0.686   0       0       0       0    
    ##  7 A                37.7 LOR1I_…      41   0       0       0       0       0    
    ##  8 A                37.7 LOR1I_…    6242   0       0       0       0       0    
    ##  9 C                37.7 LOR1I_…    6417   0       0       0       0       0    
    ## 10 T                37.7 LOR1I_…    6425   0       0       0       0       0    
    ## # … with 742 more rows, 823 more variables: `HGDP00011-Brahui` <dbl>,
    ## #   `HGDP00013-Brahui` <dbl>, `HGDP00015-Brahui` <dbl>,
    ## #   `HGDP00017-Brahui` <dbl>, `HGDP00021-Brahui` <dbl>,
    ## #   `HGDP00023-Brahui` <dbl>, `HGDP00025-Brahui` <dbl>,
    ## #   `HGDP00029-Brahui` <dbl>, `HGDP00031-Brahui` <dbl>,
    ## #   `HGDP00033-Brahui` <dbl>, `HGDP00035-Brahui` <dbl>,
    ## #   `HGDP00037-Brahui` <dbl>, `HGDP00039-Brahui` <dbl>, …
